# GPU Check

In [37]:
import tensorflow as tf

gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
    for gpu in gpus:
        print("Name:", gpu.name, "Type:", gpu.device_type)
        print("Num GPUs Available: ", len(gpus))

Name: /physical_device:GPU:0 Type: GPU
Num GPUs Available:  1


- 출력결과 1은 TensorFlow가 GPU를 인식하고 있으며, 케라스를 통해 GPU를 사용할 수 있다는 것을 나타냅니다.

# 1. 데이터 준비
- 주가데이터 전처리된 모습을 보며 학습데이터셋을 어떻게 전처리하여 준비해야할지 본다.
- 코인의 데이터를 로드하여 비슷한형태로 전처리한다.

In [38]:
#pip install yfinance

In [39]:
import yfinance
import numpy as np
df = yfinance.download('AAPL','2000-1-1','2020-1-1')
df = df.drop(['Volume'],1).drop(['Adj Close'],1)

[*********************100%%**********************]  1 of 1 completed
/var/folders/69/0x25zsbs24x5sj1x4z6z0sth0000gn/T/ipykernel_69468/758213981.py:4: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  df = df.drop(['Volume'],1).drop(['Adj Close'],1)
/var/folders/69/0x25zsbs24x5sj1x4z6z0sth0000gn/T/ipykernel_69468/758213981.py:4: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  df = df.drop(['Volume'],1).drop(['Adj Close'],1)


In [40]:
display(df)
# 20년간의 애플 주식종가
# 거래량, 조정된 종가는 제거

Open       High        Low      Close
Date                                                  
2000-01-03   0.936384   1.004464   0.907924   0.999442
2000-01-04   0.966518   0.987723   0.903460   0.915179
2000-01-05   0.926339   0.987165   0.919643   0.928571
2000-01-06   0.947545   0.955357   0.848214   0.848214
2000-01-07   0.861607   0.901786   0.852679   0.888393
...               ...        ...        ...        ...
2019-12-24  71.172501  71.222504  70.730003  71.067497
2019-12-26  71.205002  72.495003  71.175003  72.477501
2019-12-27  72.779999  73.492500  72.029999  72.449997
2019-12-30  72.364998  73.172501  71.305000  72.879997
2019-12-31  72.482498  73.419998  72.379997  73.412498

[5031 rows x 4 columns]

In [41]:
df2 = df.copy()
display(df2)
cols = df2.columns.tolist()
print(cols)
colname = [0]*len(cols)
print(colname)

Open       High        Low      Close
Date                                                  
2000-01-03   0.936384   1.004464   0.907924   0.999442
2000-01-04   0.966518   0.987723   0.903460   0.915179
2000-01-05   0.926339   0.987165   0.919643   0.928571
2000-01-06   0.947545   0.955357   0.848214   0.848214
2000-01-07   0.861607   0.901786   0.852679   0.888393
...               ...        ...        ...        ...
2019-12-24  71.172501  71.222504  70.730003  71.067497
2019-12-26  71.205002  72.495003  71.175003  72.477501
2019-12-27  72.779999  73.492500  72.029999  72.449997
2019-12-30  72.364998  73.172501  71.305000  72.879997
2019-12-31  72.482498  73.419998  72.379997  73.412498

[5031 rows x 4 columns]

['Open', 'High', 'Low', 'Close']
[0, 0, 0, 0]


In [42]:
print(df2.iloc[:,0].values)

[ 0.93638402  0.96651798  0.92633897 ... 72.77999878 72.36499786
 72.48249817]


# 2. 데이터셋 정규화
- 데이터 셋 정규화 : 모든 열의 이름을 숫자로 바꿈
- dataset : 모든 수치를 0과 1사이로 변환
- minmax : 각 컬럼의 최소/최대값 추출

## 공부) 정규화의 이유(목적)
1. 학습 속도 향상  
정규화를 통해 모든 입력피쳐(특성)들의 값을 0과 1 사이의 범위로 조정하면, 이는 네트워크의 가중치(weight)를 더 빠르고 효율적으로 조정할 수 있도록 도와줍니다. 가중치 최적화 과정이 더 원활하게 진행될 수 있어 학습 속도가 향상됩니다.

2. 그래디언트 소실/폭발 방지  
딥러닝 모델, 특히 깊은 신경망에서는 그래디언트 소실 또는 폭발 문제가 발생할 수 있습니다. 이는 가중치 업데이트가 제대로 이루어지지 않아 학습이 제대로 진행되지 않는 원인이 됩니다. 정규화를 통해 입력 데이터의 범위를 제한함으로써 이러한 문제를 완화할 수 있습니다.

3. 피쳐(특성) 간 균형 유지  
서로 다른 단위나 범위를 갖는 특성들을 같은 척도로 맞춤으로써, 모든 특성이 학습에 동등하게 기여할 수 있게 됩니다. 예를 들어, 한 특성이 0에서 1000의 범위를 가지고 다른 특성이 0에서 1의 범위를 가질 때, 이를 정규화하지 않으면 상대적으로 더 큰 범위를 가진 특성이 학습에 지나치게 영향을 미칠 수 있습니다.

4. 학습 안정성 증가  
데이터의 정규화는 학습 과정을 안정화시키는 데 도움이 됩니다. 입력 데이터의 분포가 너무 극단적이거나 불규칙할 경우, 학습 과정에서 불안정하거나 예측이 어려워질 수 있습니다.

따라서, 정규화는 딥러닝 모델의 성능을 향상시키고, 학습 과정을 보다 안정적이고 효율적으로 만드는 중요한 단계입니다.

In [43]:
def normalize_data(dataset):
    cols = dataset.columns.tolist()
    # 편의상 모든컬럼명을 인덱스(실수)로 바꿈
    col_name = [0]*len(cols)          # [0, 0, 0, 0]
    for i in range(len(cols)):
        col_name[i] = i               # [0, 1, 2, 3]
    dataset.columns = col_name
    dtypes = dataset.dtypes.tolist()
    minmax = list()
    for column in dataset:
        dataset = dataset.astype({column: 'float32'})
    # 각 컬럼의 min, max 저장(정규화를 위함)
    for i in range(len(cols)):
        col_values = dataset[col_name[i]]
        value_min = min(col_values)
        value_max = max(col_values)
        minmax.append([value_min, value_max])
    # 0~1사이값으로 정규화
    for column in dataset:
        values = dataset[column].values
        for i in range(len(values)):
            values[i] = (values[i] - minmax[column][0]) / (minmax[column][1] - minmax[column][0])
        dataset[column] = values
    dataset[column] = values
    return dataset,minmax
    

dataset, minmax = normalize_data(df)
values = dataset.values
display("< values > ", values, values.shape)
display("< df >", df, "< dataset >", dataset)
print('minmax\n',minmax)


'< values > '

array([[0.0097097 , 0.01049631, 0.00943526, 0.01045606],
       [0.01012507, 0.01026779, 0.00937339, 0.00930459],
       [0.00957124, 0.01026017, 0.00959768, 0.00948759],
       ...,
       [1.        , 1.        , 0.9951492 , 0.98684716],
       [0.9942796 , 0.9956318 , 0.9851011 , 0.9927232 ],
       [0.99589926, 0.9990103 , 1.        , 1.        ]], dtype=float32)

(5031, 4)

'< df >'

0          1          2          3
Date                                                  
2000-01-03   0.936384   1.004464   0.907924   0.999442
2000-01-04   0.966518   0.987723   0.903460   0.915179
2000-01-05   0.926339   0.987165   0.919643   0.928571
2000-01-06   0.947545   0.955357   0.848214   0.848214
2000-01-07   0.861607   0.901786   0.852679   0.888393
...               ...        ...        ...        ...
2019-12-24  71.172501  71.222504  70.730003  71.067497
2019-12-26  71.205002  72.495003  71.175003  72.477501
2019-12-27  72.779999  73.492500  72.029999  72.449997
2019-12-30  72.364998  73.172501  71.305000  72.879997
2019-12-31  72.482498  73.419998  72.379997  73.412498

[5031 rows x 4 columns]

'< dataset >'

0         1         2         3
Date                                              
2000-01-03  0.009710  0.010496  0.009435  0.010456
2000-01-04  0.010125  0.010268  0.009373  0.009305
2000-01-05  0.009571  0.010260  0.009598  0.009488
2000-01-06  0.009864  0.009826  0.008608  0.008389
2000-01-07  0.008679  0.009095  0.008670  0.008939
...              ...       ...       ...       ...
2019-12-24  0.977842  0.969013  0.977132  0.967955
2019-12-26  0.978290  0.986384  0.983299  0.987223
2019-12-27  1.000000  1.000000  0.995149  0.986847
2019-12-30  0.994280  0.995632  0.985101  0.992723
2019-12-31  0.995899  0.999010  1.000000  1.000000

[5031 rows x 4 columns]

minmax
 [[0.23196400701999664, 72.77999877929688], [0.23553599417209625, 73.49250030517578], [0.22714300453662872, 72.37999725341797], [0.23428599536418915, 73.4124984741211]]


In [44]:
len(values)

5031

# 시퀀스를 데이터셋으로 변환
- X는 타겟밸류(종가)를 제외한 피쳐들, 타갯밸류컬럼(종가)는 y 된다.
- X : 10일분의 데이터묶음이 5021개 (4개의 컬럼)
- y : X의 각 묶음의 마지막날의 다음날데이터(4개의 컬럼)  
- X : (5021, 10, 4) : 10행,4열의 2차원행렬이 5021개 있다. (3차원)
- y : (5021, 4) : 5021개의 행과 4개의 열 (2차원)
- 최종결과 x : (5020,10,4) , y : (5020,) 
- y는 각각의 x에 대응된 결과값(최종예상종가값 = 타겟밸류)

- 원래 데이터는 5031 x 4 형태였다. 우린 종가값을 예측하기위해 과거데이터 10개(n_steps)를 참조하기로 한다.  
그래서 연속된 시계열데이터 [10행 x 4열(피쳐)] 를 하나의 데이터로하여 총 5022개(X)가 만들어진다.  
하지만 if end_ix > len(sequence)-1 : 마지막시퀀스는 생성하지않는 코드를 추가한다.  
이유 : y종가는 x주가데이터의 다음날이어야 하므로, 마지막시퀀스까지 추가하면 배열의 범위를 벗어나므로 마지막은 제거


- 마지막으로 new_y과정에서 이 하나의 [ ] 중, 마지막 요소인, 기존 y(5021,4)의 종가인 4번째 컬럼의 값들만 모아, y로 만든다.

- 오늘 주가데이터로 다음날 종가예상을 위한 코드  
X = X[:-1]  
y = y[1:]  
- 이렇게 한차례씩 밀어서 1일의 X(주가데이터)를 참조하여 다음날의 y(종가)데이터를 맵핑할 수 있도록 한다.  

- 그리하여 최종 X,y는 5020개가 된다.  
- 데이터셋 최종결과물은 4개의 열과 과거 주가데이터 10개(행)를 한쌍으로한 데이터가 총 5020일분 있는 주가데이터 X
- X에 각각 맵핑된 다음날 종가데이터 5020개가 있는 y
- X,y는 오프셋정렬하여 5021개가 아니 5020개다.


In [45]:
def split_sequences(sequence, n_steps):
    X,y = list(), list()
    for i in range(len(sequence)):
        end_ix = i + n_steps
        if end_ix > len(sequence)-1: 
            break
        seq_x, seq_y = sequence[i:end_ix], sequence[end_ix] # 오프셋 정렬
        X.append(seq_x)
        y.append(seq_y)
    return np.array(X), np.array(y)

def data_setup(n_steps, n_seq, sequence):
    X, y = split_sequences(sequence, n_steps)
    print("< X > \n",X[:2],X.shape)                   # X : 10일분의 데이터묶음이 5021개
    print("< y > \n",y[:2],y.shape)                   # y : X의 각 묶음의 마지막날의 다음날데이터(예측할 타겟)
    n_features = X.shape[2]                       # X의 피쳐(특성) 수
    X = X.reshape((len(X), n_steps, n_features))  # 형태재구성 (샘플 수, 시퀀스 길이, 특성 수)
    # 
    new_y = []
    for term in y:
        new_term = term[-1]                       # 종가 컬럼만 추출
        new_y.append(new_term)
    return X, np.array(new_y), n_features

n_steps = 10
n_seq = 10000
rel_test_len = 0.1  # 트 : 테 = 9 : 1
X,y,n_features = data_setup(n_steps, n_seq, values)
# 다음날 종가데이터와 맵핑하기 위해 한스텝씩 오프셋정렬한다.
# 이 부분을 왜 한번 더 오프셋정렬하는지 이해가안감.
X = X[:-1]
y = y[1:]

X_test, y_test =  X[:int(len(X) * rel_test_len)], y[:int(len(X) * rel_test_len)]
X_train, y_train = X[int(len(X) * rel_test_len):], y[int(len(X) * rel_test_len):]

print("< Last X >\n",X[:3], X.shape, len(X))
print(" < last y > \n",y[:3], y.shape)
print("< n_features >\n",n_features)



< X > 
 [[[0.0097097  0.01049631 0.00943526 0.01045606]
  [0.01012507 0.01026779 0.00937339 0.00930459]
  [0.00957124 0.01026017 0.00959768 0.00948759]
  [0.00986355 0.00982597 0.00860771 0.00838949]
  [0.00867898 0.0090947  0.0086696  0.00893855]
  [0.00935587 0.00924704 0.00857678 0.00872503]
  [0.00860976 0.00889664 0.00805087 0.00811497]
  [0.00849437 0.00842436 0.00755588 0.00743628]
  [0.00843091 0.00882046 0.00829836 0.00860301]
  [0.00910973 0.00924704 0.0091491  0.00905293]]

 [[0.01012507 0.01026779 0.00937339 0.00930459]
  [0.00957124 0.01026017 0.00959768 0.00948759]
  [0.00986355 0.00982597 0.00860771 0.00838949]
  [0.00867898 0.0090947  0.0086696  0.00893855]
  [0.00935587 0.00924704 0.00857678 0.00872503]
  [0.00860976 0.00889664 0.00805087 0.00811497]
  [0.00849437 0.00842436 0.00755588 0.00743628]
  [0.00843091 0.00882046 0.00829836 0.00860301]
  [0.00910973 0.00924704 0.0091491  0.00905293]
  [0.00923281 0.0097041  0.00928058 0.00947997]]] (5021, 10, 4)
< y > 
 [[0.00

# 네트워크 훈련 / 콜백 로드
- ModelCheckpoint : Keras API에서 제공하는 콜백(callback) 중 하나로, 모델을 학습하는 동안 특정 조건을 만족하는 시점에 모델의 가중치를 자동으로 저장 
  
여기서는 검증 데이터셋의 손실이 가장 낮을 때 모델의 가중치를 저장

# LSTM

In [47]:
from keras.models import Sequential
from keras.layers import Dense
from keras.layers import Flatten
from keras.layers import Dropout
from keras.layers import LSTM

model = Sequential()
model.add(LSTM(
    64,
    activation = None, 
    input_shape=(10,4),
    return_sequences = True
))
model.add(LSTM(
    32,
    activation = None, 
    return_sequences = True
))
model.add(Flatten())
model.add(Dense(100, activation = None))
model.add(Dense(1, activation = 'sigmoid')) # 최종레이어 : sigmoid
model.compile(loss='mse', optimizer = 'adam')

# 네트워크 훈련 / 콜백로드

In [48]:
import os
from keras import callbacks
epochs = 5000
verbosity = 2
dirx = '/Users/hwangseokjun/Desktop/All/2024/[딥러닝]주가예측/models'
os.chdir(dirx)
# 학습된 모델의 최적의 가중치 저장
h5 = 'LSTM_weights.h5'
checkpoint = callbacks.ModelCheckpoint(h5,
                                      monitor='val_loss',
                                      verbose=0,
                                      save_best_only=True,
                                      save_weights_only=True,
                                      mode='auto',
                                      period=1)
callback = [checkpoint]
json = 'LSTM_network.json' # 모델 구조 저장
model_json = model.to_json()
with open(json, "w") as json_file:
    json_file.write(model_json)
history = model.fit(X_train,
                   y_train,
                   epochs=epochs,
                   batch_size=len(X_train) // 4,
                   validation_data = (X_test, y_test),
                   verbose = verbosity,
                   callbacks = callback)

Epoch 1/5000


2024-01-22 20:07:39.704771: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:113] Plugin optimizer for device_type GPU is enabled.


5/5 - 2s - loss: 0.1210 - val_loss: 0.1989 - 2s/epoch - 454ms/step
Epoch 2/5000


2024-01-22 20:07:41.264291: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:113] Plugin optimizer for device_type GPU is enabled.


5/5 - 1s - loss: 0.0966 - val_loss: 0.1202 - 1s/epoch - 208ms/step
Epoch 3/5000
5/5 - 1s - loss: 0.0540 - val_loss: 0.0422 - 889ms/epoch - 178ms/step
Epoch 4/5000
5/5 - 1s - loss: 0.0133 - val_loss: 0.0046 - 981ms/epoch - 196ms/step
Epoch 5/5000
5/5 - 1s - loss: 0.0044 - val_loss: 3.1700e-04 - 1s/epoch - 203ms/step
Epoch 6/5000
5/5 - 1s - loss: 0.0071 - val_loss: 2.5271e-04 - 995ms/epoch - 199ms/step
Epoch 7/5000
5/5 - 1s - loss: 0.0061 - val_loss: 5.9254e-04 - 884ms/epoch - 177ms/step
Epoch 8/5000
5/5 - 1s - loss: 0.0030 - val_loss: 0.0021 - 958ms/epoch - 192ms/step
Epoch 9/5000
5/5 - 1s - loss: 0.0015 - val_loss: 0.0046 - 1s/epoch - 225ms/step
Epoch 10/5000
5/5 - 1s - loss: 0.0018 - val_loss: 0.0052 - 988ms/epoch - 198ms/step
Epoch 11/5000
5/5 - 1s - loss: 0.0018 - val_loss: 0.0037 - 947ms/epoch - 189ms/step
Epoch 12/5000
5/5 - 1s - loss: 0.0013 - val_loss: 0.0022 - 977ms/epoch - 195ms/step
Epoch 13/5000
5/5 - 1s - loss: 0.0011 - val_loss: 0.0015 - 948ms/epoch - 190ms/step
Epoch 14/5

5/5 - 1s - loss: 6.2218e-04 - val_loss: 1.6851e-04 - 965ms/epoch - 193ms/step
Epoch 95/5000
5/5 - 1s - loss: 4.4990e-04 - val_loss: 6.6129e-05 - 1s/epoch - 201ms/step
Epoch 96/5000
5/5 - 1s - loss: 4.9844e-04 - val_loss: 1.8426e-05 - 959ms/epoch - 192ms/step
Epoch 97/5000
5/5 - 1s - loss: 5.9487e-04 - val_loss: 1.2028e-04 - 973ms/epoch - 195ms/step
Epoch 98/5000
5/5 - 1s - loss: 3.7160e-04 - val_loss: 2.8741e-04 - 940ms/epoch - 188ms/step
Epoch 99/5000
5/5 - 1s - loss: 2.9259e-04 - val_loss: 6.1875e-05 - 854ms/epoch - 171ms/step
Epoch 100/5000
5/5 - 1s - loss: 2.2315e-04 - val_loss: 8.7323e-06 - 908ms/epoch - 182ms/step
Epoch 101/5000
5/5 - 1s - loss: 2.6375e-04 - val_loss: 7.8037e-06 - 884ms/epoch - 177ms/step
Epoch 102/5000
5/5 - 1s - loss: 2.2428e-04 - val_loss: 2.1329e-05 - 962ms/epoch - 192ms/step
Epoch 103/5000
5/5 - 1s - loss: 2.1872e-04 - val_loss: 1.6234e-05 - 956ms/epoch - 191ms/step
Epoch 104/5000
5/5 - 1s - loss: 2.0978e-04 - val_loss: 7.7649e-06 - 969ms/epoch - 194ms/step


5/5 - 1s - loss: 2.2463e-04 - val_loss: 2.2657e-05 - 1s/epoch - 202ms/step
Epoch 184/5000
5/5 - 1s - loss: 2.3839e-04 - val_loss: 4.9883e-05 - 965ms/epoch - 193ms/step
Epoch 185/5000
5/5 - 1s - loss: 2.2184e-04 - val_loss: 7.4563e-05 - 994ms/epoch - 199ms/step
Epoch 186/5000
5/5 - 1s - loss: 2.3116e-04 - val_loss: 2.9996e-05 - 994ms/epoch - 199ms/step
Epoch 187/5000
5/5 - 1s - loss: 2.5187e-04 - val_loss: 1.1072e-05 - 933ms/epoch - 187ms/step
Epoch 188/5000
5/5 - 1s - loss: 2.5752e-04 - val_loss: 1.8024e-05 - 995ms/epoch - 199ms/step
Epoch 189/5000
5/5 - 1s - loss: 2.1530e-04 - val_loss: 5.8496e-05 - 995ms/epoch - 199ms/step
Epoch 190/5000
5/5 - 1s - loss: 2.4504e-04 - val_loss: 1.1814e-05 - 1s/epoch - 209ms/step
Epoch 191/5000
5/5 - 1s - loss: 2.2028e-04 - val_loss: 1.3900e-05 - 886ms/epoch - 177ms/step
Epoch 192/5000
5/5 - 1s - loss: 2.3707e-04 - val_loss: 3.6182e-05 - 1s/epoch - 204ms/step
Epoch 193/5000
5/5 - 1s - loss: 2.6009e-04 - val_loss: 1.5781e-05 - 945ms/epoch - 189ms/step
E

Epoch 273/5000
5/5 - 1s - loss: 1.8281e-04 - val_loss: 2.6793e-05 - 989ms/epoch - 198ms/step
Epoch 274/5000
5/5 - 1s - loss: 1.8242e-04 - val_loss: 2.0471e-05 - 987ms/epoch - 197ms/step
Epoch 275/5000
5/5 - 1s - loss: 2.1089e-04 - val_loss: 2.0505e-05 - 955ms/epoch - 191ms/step
Epoch 276/5000
5/5 - 1s - loss: 1.9528e-04 - val_loss: 1.2388e-05 - 910ms/epoch - 182ms/step
Epoch 277/5000
5/5 - 1s - loss: 1.8669e-04 - val_loss: 3.5554e-05 - 930ms/epoch - 186ms/step
Epoch 278/5000
5/5 - 1s - loss: 2.3138e-04 - val_loss: 1.3426e-05 - 942ms/epoch - 188ms/step
Epoch 279/5000
5/5 - 1s - loss: 2.8925e-04 - val_loss: 3.6802e-05 - 907ms/epoch - 181ms/step
Epoch 280/5000
5/5 - 1s - loss: 2.8419e-04 - val_loss: 8.1301e-05 - 983ms/epoch - 197ms/step
Epoch 281/5000
5/5 - 1s - loss: 2.0576e-04 - val_loss: 9.0330e-06 - 978ms/epoch - 196ms/step
Epoch 282/5000
5/5 - 1s - loss: 2.0691e-04 - val_loss: 1.6302e-05 - 1s/epoch - 213ms/step
Epoch 283/5000
5/5 - 1s - loss: 2.0162e-04 - val_loss: 2.5506e-05 - 936ms

Epoch 362/5000
5/5 - 1s - loss: 1.7951e-04 - val_loss: 4.0067e-05 - 1s/epoch - 205ms/step
Epoch 363/5000
5/5 - 1s - loss: 1.6997e-04 - val_loss: 4.7712e-05 - 1s/epoch - 210ms/step
Epoch 364/5000
5/5 - 1s - loss: 1.6822e-04 - val_loss: 2.4499e-05 - 924ms/epoch - 185ms/step
Epoch 365/5000
5/5 - 1s - loss: 1.7258e-04 - val_loss: 5.0730e-05 - 993ms/epoch - 199ms/step
Epoch 366/5000
5/5 - 1s - loss: 1.7995e-04 - val_loss: 4.2101e-05 - 951ms/epoch - 190ms/step
Epoch 367/5000
5/5 - 1s - loss: 1.6943e-04 - val_loss: 1.8747e-05 - 991ms/epoch - 198ms/step
Epoch 368/5000
5/5 - 1s - loss: 1.6719e-04 - val_loss: 6.8224e-05 - 967ms/epoch - 193ms/step
Epoch 369/5000
5/5 - 1s - loss: 2.0299e-04 - val_loss: 2.2148e-05 - 970ms/epoch - 194ms/step
Epoch 370/5000
5/5 - 1s - loss: 1.7920e-04 - val_loss: 1.0640e-05 - 1s/epoch - 206ms/step
Epoch 371/5000
5/5 - 1s - loss: 2.3613e-04 - val_loss: 5.4994e-05 - 936ms/epoch - 187ms/step
Epoch 372/5000
5/5 - 1s - loss: 1.9909e-04 - val_loss: 1.1785e-04 - 1s/epoch - 

Epoch 452/5000
5/5 - 1s - loss: 2.1044e-04 - val_loss: 7.6467e-06 - 948ms/epoch - 190ms/step
Epoch 453/5000
5/5 - 1s - loss: 2.2153e-04 - val_loss: 1.1139e-05 - 994ms/epoch - 199ms/step
Epoch 454/5000
5/5 - 1s - loss: 4.5390e-04 - val_loss: 7.7768e-06 - 973ms/epoch - 195ms/step
Epoch 455/5000
5/5 - 1s - loss: 4.1876e-04 - val_loss: 1.6416e-04 - 887ms/epoch - 177ms/step
Epoch 456/5000
5/5 - 1s - loss: 2.9800e-04 - val_loss: 2.5209e-05 - 884ms/epoch - 177ms/step
Epoch 457/5000
5/5 - 1s - loss: 1.9892e-04 - val_loss: 7.2235e-06 - 933ms/epoch - 187ms/step
Epoch 458/5000
5/5 - 1s - loss: 2.4209e-04 - val_loss: 2.4770e-05 - 888ms/epoch - 178ms/step
Epoch 459/5000
5/5 - 1s - loss: 1.8059e-04 - val_loss: 2.8809e-05 - 1s/epoch - 200ms/step
Epoch 460/5000
5/5 - 1s - loss: 1.8496e-04 - val_loss: 1.2379e-04 - 984ms/epoch - 197ms/step
Epoch 461/5000
5/5 - 1s - loss: 5.2723e-04 - val_loss: 1.8903e-05 - 933ms/epoch - 187ms/step
Epoch 462/5000
5/5 - 1s - loss: 6.0663e-04 - val_loss: 7.3072e-05 - 999ms

5/5 - 1s - loss: 2.4069e-04 - val_loss: 7.4040e-06 - 1000ms/epoch - 200ms/step
Epoch 542/5000
5/5 - 1s - loss: 1.8742e-04 - val_loss: 1.3841e-04 - 915ms/epoch - 183ms/step
Epoch 543/5000
5/5 - 1s - loss: 1.8212e-04 - val_loss: 8.0245e-06 - 972ms/epoch - 194ms/step
Epoch 544/5000
5/5 - 1s - loss: 1.9902e-04 - val_loss: 6.6225e-06 - 1s/epoch - 203ms/step
Epoch 545/5000
5/5 - 1s - loss: 1.6143e-04 - val_loss: 2.3016e-05 - 965ms/epoch - 193ms/step
Epoch 546/5000
5/5 - 1s - loss: 1.5579e-04 - val_loss: 1.1936e-05 - 972ms/epoch - 194ms/step
Epoch 547/5000
5/5 - 1s - loss: 1.4456e-04 - val_loss: 2.3559e-05 - 1s/epoch - 206ms/step
Epoch 548/5000
5/5 - 1s - loss: 2.0096e-04 - val_loss: 3.1454e-05 - 976ms/epoch - 195ms/step
Epoch 549/5000
5/5 - 1s - loss: 1.6961e-04 - val_loss: 1.6241e-05 - 959ms/epoch - 192ms/step
Epoch 550/5000
5/5 - 1s - loss: 1.5943e-04 - val_loss: 6.9689e-06 - 1s/epoch - 219ms/step
Epoch 551/5000
5/5 - 1s - loss: 1.7718e-04 - val_loss: 8.8114e-06 - 946ms/epoch - 189ms/step


Epoch 631/5000
5/5 - 1s - loss: 1.5892e-04 - val_loss: 6.2173e-06 - 884ms/epoch - 177ms/step
Epoch 632/5000
5/5 - 1s - loss: 1.2679e-04 - val_loss: 2.6828e-05 - 933ms/epoch - 187ms/step
Epoch 633/5000
5/5 - 1s - loss: 1.2235e-04 - val_loss: 6.2040e-06 - 1s/epoch - 205ms/step
Epoch 634/5000
5/5 - 1s - loss: 1.2227e-04 - val_loss: 7.8071e-06 - 1s/epoch - 208ms/step
Epoch 635/5000
5/5 - 1s - loss: 1.4046e-04 - val_loss: 5.9108e-06 - 977ms/epoch - 195ms/step
Epoch 636/5000
5/5 - 1s - loss: 1.2825e-04 - val_loss: 1.2711e-05 - 907ms/epoch - 181ms/step
Epoch 637/5000
5/5 - 1s - loss: 1.2483e-04 - val_loss: 1.5179e-05 - 1s/epoch - 226ms/step
Epoch 638/5000
5/5 - 1s - loss: 1.2928e-04 - val_loss: 1.1460e-05 - 1s/epoch - 213ms/step
Epoch 639/5000
5/5 - 1s - loss: 1.1858e-04 - val_loss: 6.4011e-06 - 986ms/epoch - 197ms/step
Epoch 640/5000
5/5 - 1s - loss: 1.2015e-04 - val_loss: 5.9228e-06 - 971ms/epoch - 194ms/step
Epoch 641/5000
5/5 - 1s - loss: 1.2699e-04 - val_loss: 1.2038e-05 - 1s/epoch - 209

Epoch 722/5000
5/5 - 1s - loss: 1.4210e-04 - val_loss: 1.0044e-05 - 1s/epoch - 256ms/step
Epoch 723/5000
5/5 - 1s - loss: 1.5643e-04 - val_loss: 6.7809e-06 - 1s/epoch - 241ms/step
Epoch 724/5000
5/5 - 1s - loss: 2.0826e-04 - val_loss: 8.4286e-06 - 1s/epoch - 232ms/step
Epoch 725/5000
5/5 - 1s - loss: 1.6916e-04 - val_loss: 3.3407e-05 - 1s/epoch - 251ms/step
Epoch 726/5000
5/5 - 1s - loss: 1.6207e-04 - val_loss: 1.3313e-05 - 1s/epoch - 260ms/step
Epoch 727/5000
5/5 - 1s - loss: 1.6638e-04 - val_loss: 7.8170e-06 - 1s/epoch - 251ms/step
Epoch 728/5000
5/5 - 1s - loss: 1.5645e-04 - val_loss: 2.0534e-05 - 1s/epoch - 260ms/step
Epoch 729/5000
5/5 - 1s - loss: 2.6331e-04 - val_loss: 2.1648e-05 - 1s/epoch - 288ms/step
Epoch 730/5000
5/5 - 1s - loss: 2.7018e-04 - val_loss: 4.8649e-06 - 1s/epoch - 298ms/step
Epoch 731/5000
5/5 - 1s - loss: 2.8661e-04 - val_loss: 1.4382e-05 - 1s/epoch - 283ms/step
Epoch 732/5000
5/5 - 2s - loss: 2.3241e-04 - val_loss: 3.4059e-05 - 2s/epoch - 367ms/step
Epoch 733/

Epoch 813/5000
5/5 - 1s - loss: 1.5589e-04 - val_loss: 6.3355e-06 - 1s/epoch - 217ms/step
Epoch 814/5000
5/5 - 1s - loss: 1.3269e-04 - val_loss: 8.0516e-06 - 1s/epoch - 221ms/step
Epoch 815/5000
5/5 - 1s - loss: 1.2804e-04 - val_loss: 1.2356e-05 - 1s/epoch - 248ms/step
Epoch 816/5000
5/5 - 1s - loss: 1.3685e-04 - val_loss: 5.5488e-06 - 1s/epoch - 243ms/step
Epoch 817/5000
5/5 - 1s - loss: 1.4426e-04 - val_loss: 1.0982e-05 - 1s/epoch - 226ms/step
Epoch 818/5000
5/5 - 1s - loss: 1.1065e-04 - val_loss: 8.6772e-06 - 1s/epoch - 228ms/step
Epoch 819/5000
5/5 - 1s - loss: 1.6388e-04 - val_loss: 1.0377e-05 - 1s/epoch - 217ms/step
Epoch 820/5000
5/5 - 1s - loss: 1.3754e-04 - val_loss: 1.7043e-05 - 1s/epoch - 204ms/step
Epoch 821/5000
5/5 - 1s - loss: 1.3194e-04 - val_loss: 6.6832e-06 - 1s/epoch - 201ms/step
Epoch 822/5000
5/5 - 1s - loss: 1.1575e-04 - val_loss: 4.6728e-06 - 1s/epoch - 245ms/step
Epoch 823/5000
5/5 - 1s - loss: 1.0531e-04 - val_loss: 1.3854e-05 - 1s/epoch - 221ms/step
Epoch 824/

Epoch 904/5000
5/5 - 1s - loss: 1.8051e-04 - val_loss: 7.6246e-06 - 1s/epoch - 212ms/step
Epoch 905/5000
5/5 - 1s - loss: 1.5970e-04 - val_loss: 6.2099e-05 - 1s/epoch - 238ms/step
Epoch 906/5000
5/5 - 1s - loss: 1.5101e-04 - val_loss: 5.9599e-05 - 1s/epoch - 247ms/step
Epoch 907/5000
5/5 - 1s - loss: 1.2296e-04 - val_loss: 9.3870e-06 - 1s/epoch - 228ms/step
Epoch 908/5000
5/5 - 1s - loss: 1.6104e-04 - val_loss: 1.0598e-05 - 1s/epoch - 202ms/step
Epoch 909/5000
5/5 - 1s - loss: 1.1187e-04 - val_loss: 2.0345e-05 - 1s/epoch - 227ms/step
Epoch 910/5000
5/5 - 1s - loss: 1.1082e-04 - val_loss: 4.4325e-06 - 1s/epoch - 241ms/step
Epoch 911/5000
5/5 - 1s - loss: 1.9660e-04 - val_loss: 5.2529e-06 - 1s/epoch - 228ms/step
Epoch 912/5000
5/5 - 1s - loss: 1.2040e-04 - val_loss: 5.4882e-06 - 1s/epoch - 208ms/step
Epoch 913/5000
5/5 - 1s - loss: 2.0738e-04 - val_loss: 5.6069e-06 - 1s/epoch - 211ms/step
Epoch 914/5000
5/5 - 1s - loss: 1.1513e-04 - val_loss: 6.9101e-06 - 1s/epoch - 221ms/step
Epoch 915/

Epoch 995/5000
5/5 - 1s - loss: 1.1825e-04 - val_loss: 8.5356e-06 - 1s/epoch - 209ms/step
Epoch 996/5000
5/5 - 1s - loss: 1.3441e-04 - val_loss: 7.0701e-06 - 1s/epoch - 214ms/step
Epoch 997/5000
5/5 - 1s - loss: 1.1671e-04 - val_loss: 1.8646e-05 - 1s/epoch - 206ms/step
Epoch 998/5000
5/5 - 1s - loss: 9.8488e-05 - val_loss: 7.2953e-06 - 1s/epoch - 213ms/step
Epoch 999/5000
5/5 - 1s - loss: 9.7024e-05 - val_loss: 5.5253e-06 - 1s/epoch - 215ms/step
Epoch 1000/5000
5/5 - 1s - loss: 1.0516e-04 - val_loss: 5.2131e-06 - 1000ms/epoch - 200ms/step
Epoch 1001/5000
5/5 - 1s - loss: 8.6206e-05 - val_loss: 5.3186e-06 - 1s/epoch - 205ms/step
Epoch 1002/5000
5/5 - 1s - loss: 8.5812e-05 - val_loss: 4.4774e-06 - 1s/epoch - 205ms/step
Epoch 1003/5000
5/5 - 1s - loss: 9.4751e-05 - val_loss: 4.1779e-06 - 1s/epoch - 216ms/step
Epoch 1004/5000
5/5 - 1s - loss: 8.5294e-05 - val_loss: 5.6462e-06 - 988ms/epoch - 198ms/step
Epoch 1005/5000
5/5 - 1s - loss: 8.9697e-05 - val_loss: 7.5129e-06 - 1s/epoch - 202ms/st

Epoch 1084/5000
5/5 - 1s - loss: 1.2307e-04 - val_loss: 5.1286e-06 - 902ms/epoch - 180ms/step
Epoch 1085/5000
5/5 - 1s - loss: 1.0081e-04 - val_loss: 1.8734e-05 - 980ms/epoch - 196ms/step
Epoch 1086/5000
5/5 - 1s - loss: 1.3032e-04 - val_loss: 6.8665e-06 - 973ms/epoch - 195ms/step
Epoch 1087/5000
5/5 - 1s - loss: 1.2012e-04 - val_loss: 4.3937e-06 - 945ms/epoch - 189ms/step
Epoch 1088/5000
5/5 - 1s - loss: 1.0002e-04 - val_loss: 1.4111e-05 - 1s/epoch - 201ms/step
Epoch 1089/5000
5/5 - 1s - loss: 8.9258e-05 - val_loss: 4.1234e-06 - 1s/epoch - 207ms/step
Epoch 1090/5000
5/5 - 1s - loss: 8.6922e-05 - val_loss: 4.3426e-06 - 1s/epoch - 201ms/step
Epoch 1091/5000
5/5 - 1s - loss: 8.7414e-05 - val_loss: 3.9989e-06 - 1s/epoch - 207ms/step
Epoch 1092/5000
5/5 - 1s - loss: 8.5283e-05 - val_loss: 6.6824e-06 - 906ms/epoch - 181ms/step
Epoch 1093/5000
5/5 - 1s - loss: 1.1607e-04 - val_loss: 8.6505e-06 - 911ms/epoch - 182ms/step
Epoch 1094/5000
5/5 - 1s - loss: 1.1073e-04 - val_loss: 4.1165e-06 - 986

Epoch 1173/5000
5/5 - 1s - loss: 1.0344e-04 - val_loss: 4.6683e-06 - 1s/epoch - 210ms/step
Epoch 1174/5000
5/5 - 1s - loss: 9.5421e-05 - val_loss: 9.9247e-06 - 1s/epoch - 217ms/step
Epoch 1175/5000
5/5 - 1s - loss: 9.4736e-05 - val_loss: 3.3070e-05 - 1s/epoch - 212ms/step
Epoch 1176/5000
5/5 - 1s - loss: 1.9888e-04 - val_loss: 5.6885e-06 - 1s/epoch - 222ms/step
Epoch 1177/5000
5/5 - 1s - loss: 1.2046e-04 - val_loss: 9.3129e-06 - 1s/epoch - 233ms/step
Epoch 1178/5000
5/5 - 1s - loss: 1.1382e-04 - val_loss: 1.1627e-05 - 1s/epoch - 207ms/step
Epoch 1179/5000
5/5 - 1s - loss: 1.0237e-04 - val_loss: 4.4416e-06 - 1s/epoch - 203ms/step
Epoch 1180/5000
5/5 - 1s - loss: 1.2101e-04 - val_loss: 8.1606e-06 - 1s/epoch - 213ms/step
Epoch 1181/5000
5/5 - 1s - loss: 8.2930e-05 - val_loss: 4.6299e-06 - 1s/epoch - 220ms/step
Epoch 1182/5000
5/5 - 1s - loss: 8.6683e-05 - val_loss: 5.2795e-06 - 1s/epoch - 217ms/step
Epoch 1183/5000
5/5 - 1s - loss: 8.6433e-05 - val_loss: 7.0088e-06 - 1s/epoch - 236ms/step

Epoch 1263/5000
5/5 - 1s - loss: 1.0075e-04 - val_loss: 3.2972e-06 - 1s/epoch - 211ms/step
Epoch 1264/5000
5/5 - 1s - loss: 1.0351e-04 - val_loss: 4.6538e-06 - 1s/epoch - 202ms/step
Epoch 1265/5000
5/5 - 1s - loss: 8.8863e-05 - val_loss: 4.5856e-06 - 1s/epoch - 201ms/step
Epoch 1266/5000
5/5 - 1s - loss: 8.9528e-05 - val_loss: 7.4675e-06 - 926ms/epoch - 185ms/step
Epoch 1267/5000
5/5 - 1s - loss: 8.3880e-05 - val_loss: 3.6308e-06 - 971ms/epoch - 194ms/step
Epoch 1268/5000
5/5 - 1s - loss: 9.1261e-05 - val_loss: 4.2236e-06 - 928ms/epoch - 186ms/step
Epoch 1269/5000
5/5 - 1s - loss: 8.8659e-05 - val_loss: 4.0114e-06 - 923ms/epoch - 185ms/step
Epoch 1270/5000
5/5 - 1s - loss: 9.3875e-05 - val_loss: 3.8129e-06 - 971ms/epoch - 194ms/step
Epoch 1271/5000
5/5 - 1s - loss: 8.4170e-05 - val_loss: 6.2776e-06 - 1s/epoch - 202ms/step
Epoch 1272/5000
5/5 - 1s - loss: 8.2576e-05 - val_loss: 6.0531e-06 - 980ms/epoch - 196ms/step
Epoch 1273/5000
5/5 - 1s - loss: 7.9497e-05 - val_loss: 1.2208e-05 - 1s/

Epoch 1352/5000
5/5 - 1s - loss: 8.6065e-05 - val_loss: 8.8395e-06 - 1s/epoch - 203ms/step
Epoch 1353/5000
5/5 - 1s - loss: 8.3685e-05 - val_loss: 3.8952e-06 - 1s/epoch - 215ms/step
Epoch 1354/5000
5/5 - 1s - loss: 1.0980e-04 - val_loss: 3.4238e-06 - 1s/epoch - 223ms/step
Epoch 1355/5000
5/5 - 1s - loss: 7.9977e-05 - val_loss: 5.2689e-06 - 1s/epoch - 220ms/step
Epoch 1356/5000
5/5 - 1s - loss: 9.4260e-05 - val_loss: 3.4097e-06 - 1s/epoch - 210ms/step
Epoch 1357/5000
5/5 - 1s - loss: 8.6644e-05 - val_loss: 3.8728e-06 - 1s/epoch - 220ms/step
Epoch 1358/5000
5/5 - 1s - loss: 7.7299e-05 - val_loss: 9.9883e-06 - 979ms/epoch - 196ms/step
Epoch 1359/5000
5/5 - 1s - loss: 2.5147e-04 - val_loss: 1.1037e-05 - 1s/epoch - 204ms/step
Epoch 1360/5000
5/5 - 1s - loss: 1.2277e-04 - val_loss: 1.5986e-05 - 1s/epoch - 200ms/step
Epoch 1361/5000
5/5 - 1s - loss: 1.2397e-04 - val_loss: 1.2470e-05 - 1s/epoch - 210ms/step
Epoch 1362/5000
5/5 - 1s - loss: 1.5358e-04 - val_loss: 1.2021e-05 - 995ms/epoch - 199m

Epoch 1441/5000
5/5 - 1s - loss: 8.5632e-05 - val_loss: 3.6938e-06 - 890ms/epoch - 178ms/step
Epoch 1442/5000
5/5 - 1s - loss: 2.0429e-04 - val_loss: 3.8567e-06 - 1s/epoch - 217ms/step
Epoch 1443/5000
5/5 - 1s - loss: 1.0496e-04 - val_loss: 6.5068e-06 - 957ms/epoch - 191ms/step
Epoch 1444/5000
5/5 - 1s - loss: 1.9174e-04 - val_loss: 1.5449e-05 - 1s/epoch - 200ms/step
Epoch 1445/5000
5/5 - 1s - loss: 1.4279e-04 - val_loss: 8.8662e-06 - 921ms/epoch - 184ms/step
Epoch 1446/5000
5/5 - 1s - loss: 1.2225e-04 - val_loss: 3.8988e-05 - 966ms/epoch - 193ms/step
Epoch 1447/5000
5/5 - 1s - loss: 1.1690e-04 - val_loss: 4.0235e-05 - 959ms/epoch - 192ms/step
Epoch 1448/5000
5/5 - 1s - loss: 9.2552e-05 - val_loss: 3.5945e-06 - 900ms/epoch - 180ms/step
Epoch 1449/5000
5/5 - 1s - loss: 8.3780e-05 - val_loss: 8.5748e-06 - 1s/epoch - 203ms/step
Epoch 1450/5000
5/5 - 1s - loss: 1.9086e-04 - val_loss: 4.2241e-06 - 969ms/epoch - 194ms/step
Epoch 1451/5000
5/5 - 1s - loss: 1.4810e-04 - val_loss: 3.9954e-06 - 

5/5 - 1s - loss: 1.5997e-04 - val_loss: 1.0349e-04 - 973ms/epoch - 195ms/step
Epoch 1530/5000
5/5 - 1s - loss: 1.8299e-04 - val_loss: 4.4549e-06 - 964ms/epoch - 193ms/step
Epoch 1531/5000
5/5 - 1s - loss: 1.0636e-04 - val_loss: 1.4951e-05 - 955ms/epoch - 191ms/step
Epoch 1532/5000
5/5 - 1s - loss: 1.0386e-04 - val_loss: 3.7392e-06 - 906ms/epoch - 181ms/step
Epoch 1533/5000
5/5 - 1s - loss: 9.4015e-05 - val_loss: 4.0610e-06 - 899ms/epoch - 180ms/step
Epoch 1534/5000
5/5 - 1s - loss: 1.8630e-04 - val_loss: 2.1168e-05 - 982ms/epoch - 196ms/step
Epoch 1535/5000
5/5 - 1s - loss: 2.2623e-04 - val_loss: 3.4198e-06 - 1s/epoch - 204ms/step
Epoch 1536/5000
5/5 - 1s - loss: 2.8563e-04 - val_loss: 2.3937e-05 - 960ms/epoch - 192ms/step
Epoch 1537/5000
5/5 - 1s - loss: 1.9974e-04 - val_loss: 5.3860e-06 - 990ms/epoch - 198ms/step
Epoch 1538/5000
5/5 - 1s - loss: 1.0242e-04 - val_loss: 1.1665e-04 - 976ms/epoch - 195ms/step
Epoch 1539/5000
5/5 - 1s - loss: 1.2948e-04 - val_loss: 3.1636e-05 - 946ms/epoc

Epoch 1618/5000
5/5 - 1s - loss: 6.9993e-05 - val_loss: 3.1663e-06 - 993ms/epoch - 199ms/step
Epoch 1619/5000
5/5 - 1s - loss: 7.0288e-05 - val_loss: 3.3870e-06 - 1s/epoch - 208ms/step
Epoch 1620/5000
5/5 - 1s - loss: 7.6037e-05 - val_loss: 7.2549e-06 - 1s/epoch - 209ms/step
Epoch 1621/5000
5/5 - 1s - loss: 7.3821e-05 - val_loss: 2.8527e-06 - 1s/epoch - 206ms/step
Epoch 1622/5000
5/5 - 1s - loss: 1.0466e-04 - val_loss: 9.8234e-06 - 1s/epoch - 209ms/step
Epoch 1623/5000
5/5 - 1s - loss: 1.1775e-04 - val_loss: 4.3619e-06 - 926ms/epoch - 185ms/step
Epoch 1624/5000
5/5 - 1s - loss: 1.1368e-04 - val_loss: 3.3665e-06 - 1s/epoch - 206ms/step
Epoch 1625/5000
5/5 - 1s - loss: 7.9319e-05 - val_loss: 8.1858e-06 - 940ms/epoch - 188ms/step
Epoch 1626/5000
5/5 - 1s - loss: 7.8254e-05 - val_loss: 3.0379e-06 - 911ms/epoch - 182ms/step
Epoch 1627/5000
5/5 - 1s - loss: 1.0039e-04 - val_loss: 2.7529e-06 - 900ms/epoch - 180ms/step
Epoch 1628/5000
5/5 - 1s - loss: 7.9951e-05 - val_loss: 3.8097e-06 - 1s/epo

Epoch 1707/5000
5/5 - 1s - loss: 6.9074e-05 - val_loss: 3.3216e-06 - 1s/epoch - 201ms/step
Epoch 1708/5000
5/5 - 1s - loss: 6.9036e-05 - val_loss: 1.2091e-05 - 993ms/epoch - 199ms/step
Epoch 1709/5000
5/5 - 1s - loss: 7.1364e-05 - val_loss: 7.7661e-06 - 1s/epoch - 204ms/step
Epoch 1710/5000
5/5 - 1s - loss: 7.3206e-05 - val_loss: 4.0030e-06 - 949ms/epoch - 190ms/step
Epoch 1711/5000
5/5 - 1s - loss: 7.2817e-05 - val_loss: 3.2346e-06 - 1s/epoch - 209ms/step
Epoch 1712/5000
5/5 - 1s - loss: 7.1570e-05 - val_loss: 5.0398e-06 - 948ms/epoch - 190ms/step
Epoch 1713/5000
5/5 - 1s - loss: 6.7488e-05 - val_loss: 3.2428e-06 - 1s/epoch - 201ms/step
Epoch 1714/5000
5/5 - 1s - loss: 7.3279e-05 - val_loss: 4.0242e-06 - 1000ms/epoch - 200ms/step
Epoch 1715/5000
5/5 - 1s - loss: 1.0097e-04 - val_loss: 5.2748e-06 - 1s/epoch - 204ms/step
Epoch 1716/5000
5/5 - 1s - loss: 1.2523e-04 - val_loss: 4.7463e-06 - 1s/epoch - 217ms/step
Epoch 1717/5000
5/5 - 1s - loss: 9.6481e-05 - val_loss: 2.9636e-06 - 991ms/ep

Epoch 1796/5000
5/5 - 1s - loss: 1.2210e-04 - val_loss: 3.7810e-06 - 885ms/epoch - 177ms/step
Epoch 1797/5000
5/5 - 1s - loss: 1.0724e-04 - val_loss: 6.2274e-06 - 918ms/epoch - 184ms/step
Epoch 1798/5000
5/5 - 1s - loss: 1.1641e-04 - val_loss: 1.8057e-05 - 966ms/epoch - 193ms/step
Epoch 1799/5000
5/5 - 1s - loss: 1.1596e-04 - val_loss: 3.7080e-06 - 981ms/epoch - 196ms/step
Epoch 1800/5000
5/5 - 1s - loss: 1.0866e-04 - val_loss: 6.9877e-06 - 915ms/epoch - 183ms/step
Epoch 1801/5000
5/5 - 1s - loss: 8.4921e-05 - val_loss: 3.1899e-06 - 916ms/epoch - 183ms/step
Epoch 1802/5000
5/5 - 1s - loss: 9.9825e-05 - val_loss: 5.5097e-06 - 1s/epoch - 210ms/step
Epoch 1803/5000
5/5 - 1s - loss: 9.7677e-05 - val_loss: 8.9711e-06 - 999ms/epoch - 200ms/step
Epoch 1804/5000
5/5 - 1s - loss: 8.4046e-05 - val_loss: 9.2128e-06 - 890ms/epoch - 178ms/step
Epoch 1805/5000
5/5 - 1s - loss: 7.5647e-05 - val_loss: 3.9469e-06 - 943ms/epoch - 189ms/step
Epoch 1806/5000
5/5 - 1s - loss: 8.1120e-05 - val_loss: 3.6201e

5/5 - 1s - loss: 7.2071e-05 - val_loss: 2.7201e-06 - 976ms/epoch - 195ms/step
Epoch 1885/5000
5/5 - 1s - loss: 7.4725e-05 - val_loss: 6.2027e-06 - 938ms/epoch - 188ms/step
Epoch 1886/5000
5/5 - 1s - loss: 7.3498e-05 - val_loss: 3.1236e-06 - 950ms/epoch - 190ms/step
Epoch 1887/5000
5/5 - 1s - loss: 6.6959e-05 - val_loss: 7.3331e-06 - 985ms/epoch - 197ms/step
Epoch 1888/5000
5/5 - 1s - loss: 6.5649e-05 - val_loss: 3.1267e-06 - 995ms/epoch - 199ms/step
Epoch 1889/5000
5/5 - 1s - loss: 6.8951e-05 - val_loss: 3.0052e-06 - 920ms/epoch - 184ms/step
Epoch 1890/5000
5/5 - 1s - loss: 6.4825e-05 - val_loss: 4.0900e-06 - 931ms/epoch - 186ms/step
Epoch 1891/5000
5/5 - 1s - loss: 1.1305e-04 - val_loss: 4.0627e-06 - 1s/epoch - 215ms/step
Epoch 1892/5000
5/5 - 1s - loss: 7.1690e-05 - val_loss: 3.4337e-06 - 1s/epoch - 207ms/step
Epoch 1893/5000
5/5 - 1s - loss: 6.9726e-05 - val_loss: 3.4246e-06 - 920ms/epoch - 184ms/step
Epoch 1894/5000
5/5 - 1s - loss: 7.1484e-05 - val_loss: 3.5233e-06 - 942ms/epoch -

5/5 - 1s - loss: 1.5207e-04 - val_loss: 1.4377e-05 - 949ms/epoch - 190ms/step
Epoch 1973/5000
5/5 - 1s - loss: 7.0956e-05 - val_loss: 5.8663e-06 - 889ms/epoch - 178ms/step
Epoch 1974/5000
5/5 - 1s - loss: 9.8385e-05 - val_loss: 3.1764e-06 - 904ms/epoch - 181ms/step
Epoch 1975/5000
5/5 - 1s - loss: 7.0437e-05 - val_loss: 4.9083e-06 - 877ms/epoch - 175ms/step
Epoch 1976/5000
5/5 - 1s - loss: 7.0376e-05 - val_loss: 2.6922e-06 - 970ms/epoch - 194ms/step
Epoch 1977/5000
5/5 - 1s - loss: 7.2903e-05 - val_loss: 6.6855e-06 - 1s/epoch - 216ms/step
Epoch 1978/5000
5/5 - 1s - loss: 8.1421e-05 - val_loss: 1.0139e-05 - 982ms/epoch - 196ms/step
Epoch 1979/5000
5/5 - 1s - loss: 8.3424e-05 - val_loss: 2.9906e-06 - 980ms/epoch - 196ms/step
Epoch 1980/5000
5/5 - 1s - loss: 9.0469e-05 - val_loss: 3.9598e-06 - 1s/epoch - 210ms/step
Epoch 1981/5000
5/5 - 1s - loss: 1.0114e-04 - val_loss: 2.6440e-05 - 980ms/epoch - 196ms/step
Epoch 1982/5000
5/5 - 1s - loss: 8.5494e-05 - val_loss: 2.6730e-06 - 906ms/epoch -

5/5 - 1s - loss: 7.8680e-05 - val_loss: 3.3353e-06 - 962ms/epoch - 192ms/step
Epoch 2061/5000
5/5 - 1s - loss: 1.7410e-04 - val_loss: 4.6657e-06 - 964ms/epoch - 193ms/step
Epoch 2062/5000
5/5 - 1s - loss: 8.0439e-05 - val_loss: 6.9073e-06 - 957ms/epoch - 191ms/step
Epoch 2063/5000
5/5 - 1s - loss: 8.4932e-05 - val_loss: 2.2120e-05 - 1s/epoch - 202ms/step
Epoch 2064/5000
5/5 - 1s - loss: 1.0727e-04 - val_loss: 6.8229e-06 - 989ms/epoch - 198ms/step
Epoch 2065/5000
5/5 - 1s - loss: 1.0337e-04 - val_loss: 2.8616e-06 - 1s/epoch - 202ms/step
Epoch 2066/5000
5/5 - 1s - loss: 8.7798e-05 - val_loss: 6.1782e-06 - 996ms/epoch - 199ms/step
Epoch 2067/5000
5/5 - 1s - loss: 6.4785e-05 - val_loss: 3.2561e-06 - 895ms/epoch - 179ms/step
Epoch 2068/5000
5/5 - 1s - loss: 6.9242e-05 - val_loss: 7.5911e-06 - 1s/epoch - 205ms/step
Epoch 2069/5000
5/5 - 1s - loss: 6.9133e-05 - val_loss: 2.8904e-06 - 1s/epoch - 204ms/step
Epoch 2070/5000
5/5 - 1s - loss: 6.2388e-05 - val_loss: 2.8691e-06 - 963ms/epoch - 193ms

5/5 - 1s - loss: 6.5839e-05 - val_loss: 8.0770e-06 - 1s/epoch - 212ms/step
Epoch 2149/5000
5/5 - 1s - loss: 6.4583e-05 - val_loss: 5.0033e-06 - 917ms/epoch - 183ms/step
Epoch 2150/5000
5/5 - 1s - loss: 6.4742e-05 - val_loss: 3.2232e-06 - 934ms/epoch - 187ms/step
Epoch 2151/5000
5/5 - 1s - loss: 7.9007e-05 - val_loss: 3.1802e-06 - 975ms/epoch - 195ms/step
Epoch 2152/5000
5/5 - 1s - loss: 6.5327e-05 - val_loss: 6.0783e-06 - 952ms/epoch - 190ms/step
Epoch 2153/5000
5/5 - 1s - loss: 6.6594e-05 - val_loss: 2.9295e-06 - 1s/epoch - 211ms/step
Epoch 2154/5000
5/5 - 1s - loss: 6.2867e-05 - val_loss: 2.5503e-06 - 944ms/epoch - 189ms/step
Epoch 2155/5000
5/5 - 1s - loss: 9.7891e-05 - val_loss: 3.6984e-06 - 900ms/epoch - 180ms/step
Epoch 2156/5000
5/5 - 1s - loss: 9.3699e-05 - val_loss: 5.0539e-06 - 924ms/epoch - 185ms/step
Epoch 2157/5000
5/5 - 1s - loss: 8.4324e-05 - val_loss: 3.0126e-06 - 1s/epoch - 201ms/step
Epoch 2158/5000
5/5 - 1s - loss: 7.0325e-05 - val_loss: 1.1527e-05 - 985ms/epoch - 19

Epoch 2237/5000
5/5 - 1s - loss: 8.0784e-05 - val_loss: 2.5363e-06 - 969ms/epoch - 194ms/step
Epoch 2238/5000
5/5 - 1s - loss: 6.6823e-05 - val_loss: 2.7663e-06 - 1s/epoch - 201ms/step
Epoch 2239/5000
5/5 - 1s - loss: 6.4305e-05 - val_loss: 2.6050e-06 - 1s/epoch - 204ms/step
Epoch 2240/5000
5/5 - 1s - loss: 5.7319e-05 - val_loss: 3.7832e-06 - 997ms/epoch - 199ms/step
Epoch 2241/5000
5/5 - 1s - loss: 5.8814e-05 - val_loss: 4.4430e-06 - 964ms/epoch - 193ms/step
Epoch 2242/5000
5/5 - 1s - loss: 9.2905e-05 - val_loss: 2.4741e-06 - 1s/epoch - 209ms/step
Epoch 2243/5000
5/5 - 1s - loss: 6.9244e-05 - val_loss: 3.7488e-06 - 1s/epoch - 204ms/step
Epoch 2244/5000
5/5 - 1s - loss: 7.9937e-05 - val_loss: 1.5140e-05 - 1s/epoch - 202ms/step
Epoch 2245/5000
5/5 - 1s - loss: 1.4396e-04 - val_loss: 3.0655e-06 - 979ms/epoch - 196ms/step
Epoch 2246/5000
5/5 - 1s - loss: 8.9413e-05 - val_loss: 5.8862e-06 - 969ms/epoch - 194ms/step
Epoch 2247/5000
5/5 - 1s - loss: 1.1402e-04 - val_loss: 3.6183e-06 - 923ms/

Epoch 2325/5000
5/5 - 1s - loss: 6.3584e-05 - val_loss: 3.7121e-06 - 1s/epoch - 202ms/step
Epoch 2326/5000
5/5 - 1s - loss: 7.0044e-05 - val_loss: 5.8802e-06 - 1s/epoch - 211ms/step
Epoch 2327/5000
5/5 - 1s - loss: 8.8762e-05 - val_loss: 2.4745e-06 - 946ms/epoch - 189ms/step
Epoch 2328/5000
5/5 - 1s - loss: 7.0237e-05 - val_loss: 3.0626e-06 - 986ms/epoch - 197ms/step
Epoch 2329/5000
5/5 - 1s - loss: 5.9444e-05 - val_loss: 2.2746e-06 - 888ms/epoch - 178ms/step
Epoch 2330/5000
5/5 - 1s - loss: 5.9257e-05 - val_loss: 2.6614e-06 - 918ms/epoch - 184ms/step
Epoch 2331/5000
5/5 - 1s - loss: 5.7361e-05 - val_loss: 2.7515e-06 - 930ms/epoch - 186ms/step
Epoch 2332/5000
5/5 - 1s - loss: 7.7565e-05 - val_loss: 3.8184e-06 - 950ms/epoch - 190ms/step
Epoch 2333/5000
5/5 - 1s - loss: 9.1612e-05 - val_loss: 2.3748e-06 - 887ms/epoch - 177ms/step
Epoch 2334/5000
5/5 - 1s - loss: 1.9057e-04 - val_loss: 7.0727e-06 - 908ms/epoch - 182ms/step
Epoch 2335/5000
5/5 - 1s - loss: 8.0649e-05 - val_loss: 3.3929e-06

Epoch 2414/5000
5/5 - 1s - loss: 5.8624e-05 - val_loss: 5.4287e-06 - 912ms/epoch - 182ms/step
Epoch 2415/5000
5/5 - 1s - loss: 1.3250e-04 - val_loss: 2.5609e-06 - 992ms/epoch - 198ms/step
Epoch 2416/5000
5/5 - 1s - loss: 6.4906e-05 - val_loss: 2.4858e-06 - 1s/epoch - 220ms/step
Epoch 2417/5000
5/5 - 1s - loss: 8.7872e-05 - val_loss: 5.0530e-06 - 1s/epoch - 208ms/step
Epoch 2418/5000
5/5 - 1s - loss: 6.3082e-05 - val_loss: 6.8709e-06 - 964ms/epoch - 193ms/step
Epoch 2419/5000
5/5 - 1s - loss: 8.6413e-05 - val_loss: 2.5984e-06 - 1s/epoch - 234ms/step
Epoch 2420/5000
5/5 - 1s - loss: 6.1506e-05 - val_loss: 2.9189e-06 - 1s/epoch - 226ms/step
Epoch 2421/5000
5/5 - 1s - loss: 6.7930e-05 - val_loss: 3.4243e-06 - 997ms/epoch - 199ms/step
Epoch 2422/5000
5/5 - 1s - loss: 6.3295e-05 - val_loss: 6.1608e-06 - 962ms/epoch - 192ms/step
Epoch 2423/5000
5/5 - 1s - loss: 6.5723e-05 - val_loss: 3.1393e-06 - 1s/epoch - 229ms/step
Epoch 2424/5000
5/5 - 1s - loss: 7.5676e-05 - val_loss: 2.8629e-06 - 1s/epo

Epoch 2504/5000
5/5 - 2s - loss: 5.6716e-05 - val_loss: 3.6634e-06 - 2s/epoch - 421ms/step
Epoch 2505/5000
5/5 - 2s - loss: 6.1737e-05 - val_loss: 2.3371e-06 - 2s/epoch - 325ms/step
Epoch 2506/5000
5/5 - 2s - loss: 7.3899e-05 - val_loss: 2.7228e-06 - 2s/epoch - 343ms/step
Epoch 2507/5000
5/5 - 2s - loss: 6.5932e-05 - val_loss: 5.6644e-06 - 2s/epoch - 318ms/step
Epoch 2508/5000
5/5 - 1s - loss: 7.3594e-05 - val_loss: 2.4264e-06 - 1s/epoch - 281ms/step
Epoch 2509/5000
5/5 - 2s - loss: 5.9442e-05 - val_loss: 3.6963e-06 - 2s/epoch - 306ms/step
Epoch 2510/5000
5/5 - 1s - loss: 6.3335e-05 - val_loss: 3.9304e-06 - 1s/epoch - 279ms/step
Epoch 2511/5000
5/5 - 1s - loss: 6.0146e-05 - val_loss: 2.5958e-06 - 1s/epoch - 267ms/step
Epoch 2512/5000
5/5 - 1s - loss: 9.0322e-05 - val_loss: 4.2852e-06 - 1s/epoch - 284ms/step
Epoch 2513/5000
5/5 - 1s - loss: 6.1653e-05 - val_loss: 2.2252e-06 - 1s/epoch - 274ms/step
Epoch 2514/5000
5/5 - 1s - loss: 6.0764e-05 - val_loss: 5.7033e-06 - 1s/epoch - 290ms/step

Epoch 2594/5000
5/5 - 1s - loss: 6.0934e-05 - val_loss: 4.4773e-06 - 1s/epoch - 249ms/step
Epoch 2595/5000
5/5 - 1s - loss: 8.5426e-05 - val_loss: 8.4161e-06 - 1s/epoch - 202ms/step
Epoch 2596/5000
5/5 - 1s - loss: 9.3107e-05 - val_loss: 2.7468e-06 - 1s/epoch - 218ms/step
Epoch 2597/5000
5/5 - 1s - loss: 6.1918e-05 - val_loss: 9.0698e-06 - 1s/epoch - 226ms/step
Epoch 2598/5000
5/5 - 1s - loss: 7.2885e-05 - val_loss: 4.4169e-06 - 1s/epoch - 231ms/step
Epoch 2599/5000
5/5 - 1s - loss: 6.8129e-05 - val_loss: 2.6381e-06 - 1s/epoch - 214ms/step
Epoch 2600/5000
5/5 - 1s - loss: 6.5790e-05 - val_loss: 3.8765e-06 - 983ms/epoch - 197ms/step
Epoch 2601/5000
5/5 - 1s - loss: 8.8989e-05 - val_loss: 7.9668e-06 - 1s/epoch - 224ms/step
Epoch 2602/5000
5/5 - 1s - loss: 1.3883e-04 - val_loss: 2.6499e-06 - 1s/epoch - 238ms/step
Epoch 2603/5000
5/5 - 1s - loss: 5.0554e-04 - val_loss: 4.0010e-06 - 1s/epoch - 236ms/step
Epoch 2604/5000
5/5 - 1s - loss: 6.3214e-04 - val_loss: 1.8316e-05 - 1s/epoch - 207ms/s

Epoch 2684/5000
5/5 - 1s - loss: 5.6214e-05 - val_loss: 3.3573e-06 - 1s/epoch - 205ms/step
Epoch 2685/5000
5/5 - 1s - loss: 5.4577e-05 - val_loss: 3.2768e-06 - 1s/epoch - 223ms/step
Epoch 2686/5000
5/5 - 1s - loss: 5.7924e-05 - val_loss: 1.9123e-06 - 1s/epoch - 235ms/step
Epoch 2687/5000
5/5 - 1s - loss: 5.3600e-05 - val_loss: 2.2029e-06 - 1s/epoch - 246ms/step
Epoch 2688/5000
5/5 - 1s - loss: 5.3324e-05 - val_loss: 1.9348e-06 - 1s/epoch - 215ms/step
Epoch 2689/5000
5/5 - 1s - loss: 6.0586e-05 - val_loss: 2.8528e-06 - 1s/epoch - 243ms/step
Epoch 2690/5000
5/5 - 1s - loss: 6.1812e-05 - val_loss: 2.1263e-06 - 1s/epoch - 209ms/step
Epoch 2691/5000
5/5 - 1s - loss: 5.8917e-05 - val_loss: 3.2868e-06 - 1s/epoch - 205ms/step
Epoch 2692/5000
5/5 - 1s - loss: 5.7799e-05 - val_loss: 4.6078e-06 - 1s/epoch - 202ms/step
Epoch 2693/5000
5/5 - 1s - loss: 6.2866e-05 - val_loss: 2.8469e-06 - 1s/epoch - 213ms/step
Epoch 2694/5000
5/5 - 1s - loss: 5.8546e-05 - val_loss: 2.5096e-06 - 1s/epoch - 205ms/step

Epoch 2774/5000
5/5 - 1s - loss: 6.7902e-05 - val_loss: 2.3715e-06 - 1s/epoch - 209ms/step
Epoch 2775/5000
5/5 - 1s - loss: 6.0930e-05 - val_loss: 2.1505e-06 - 1s/epoch - 213ms/step
Epoch 2776/5000
5/5 - 1s - loss: 5.6268e-05 - val_loss: 6.5165e-06 - 1s/epoch - 212ms/step
Epoch 2777/5000
5/5 - 1s - loss: 5.9772e-05 - val_loss: 2.0534e-06 - 1s/epoch - 201ms/step
Epoch 2778/5000
5/5 - 1s - loss: 5.8560e-05 - val_loss: 2.0808e-06 - 992ms/epoch - 198ms/step
Epoch 2779/5000
5/5 - 1s - loss: 5.4266e-05 - val_loss: 2.4762e-06 - 978ms/epoch - 196ms/step
Epoch 2780/5000
5/5 - 1s - loss: 5.3779e-05 - val_loss: 2.0588e-06 - 1s/epoch - 217ms/step
Epoch 2781/5000
5/5 - 1s - loss: 5.3284e-05 - val_loss: 2.0510e-06 - 1s/epoch - 207ms/step
Epoch 2782/5000
5/5 - 1s - loss: 5.9914e-05 - val_loss: 2.4940e-06 - 1s/epoch - 218ms/step
Epoch 2783/5000
5/5 - 1s - loss: 5.5460e-05 - val_loss: 2.1603e-06 - 1s/epoch - 216ms/step
Epoch 2784/5000
5/5 - 1s - loss: 5.4694e-05 - val_loss: 2.2383e-06 - 1s/epoch - 216m

Epoch 2864/5000
5/5 - 1s - loss: 5.6526e-05 - val_loss: 3.1049e-06 - 1s/epoch - 215ms/step
Epoch 2865/5000
5/5 - 1s - loss: 5.7079e-05 - val_loss: 3.0744e-06 - 1s/epoch - 203ms/step
Epoch 2866/5000
5/5 - 1s - loss: 6.9704e-05 - val_loss: 1.1714e-05 - 984ms/epoch - 197ms/step
Epoch 2867/5000
5/5 - 1s - loss: 8.1022e-05 - val_loss: 2.7905e-06 - 991ms/epoch - 198ms/step
Epoch 2868/5000
5/5 - 1s - loss: 7.5641e-05 - val_loss: 4.1287e-06 - 1s/epoch - 204ms/step
Epoch 2869/5000
5/5 - 1s - loss: 8.2040e-05 - val_loss: 7.6983e-06 - 1s/epoch - 220ms/step
Epoch 2870/5000
5/5 - 1s - loss: 8.4619e-05 - val_loss: 1.5661e-05 - 1s/epoch - 202ms/step
Epoch 2871/5000
5/5 - 1s - loss: 7.1651e-05 - val_loss: 1.7985e-06 - 1s/epoch - 216ms/step
Epoch 2872/5000
5/5 - 1s - loss: 8.7856e-05 - val_loss: 1.9168e-06 - 947ms/epoch - 189ms/step
Epoch 2873/5000
5/5 - 1s - loss: 6.1690e-05 - val_loss: 2.1237e-06 - 1s/epoch - 204ms/step
Epoch 2874/5000
5/5 - 1s - loss: 5.7136e-05 - val_loss: 2.2767e-06 - 988ms/epoch 

Epoch 2954/5000
5/5 - 1s - loss: 6.4158e-05 - val_loss: 1.9223e-06 - 1s/epoch - 224ms/step
Epoch 2955/5000
5/5 - 1s - loss: 7.8429e-05 - val_loss: 1.9433e-06 - 1s/epoch - 215ms/step
Epoch 2956/5000
5/5 - 1s - loss: 6.5439e-05 - val_loss: 3.0315e-06 - 1s/epoch - 218ms/step
Epoch 2957/5000
5/5 - 1s - loss: 5.5608e-05 - val_loss: 2.8607e-06 - 1s/epoch - 218ms/step
Epoch 2958/5000
5/5 - 1s - loss: 5.5721e-05 - val_loss: 2.4754e-06 - 1s/epoch - 224ms/step
Epoch 2959/5000
5/5 - 1s - loss: 5.5687e-05 - val_loss: 2.0246e-06 - 1s/epoch - 233ms/step
Epoch 2960/5000
5/5 - 1s - loss: 6.4710e-05 - val_loss: 4.7284e-06 - 1s/epoch - 231ms/step
Epoch 2961/5000
5/5 - 1s - loss: 6.1965e-05 - val_loss: 3.6679e-06 - 1s/epoch - 215ms/step
Epoch 2962/5000
5/5 - 1s - loss: 5.7791e-05 - val_loss: 3.3161e-06 - 1s/epoch - 213ms/step
Epoch 2963/5000
5/5 - 1s - loss: 5.4385e-05 - val_loss: 6.5597e-06 - 1s/epoch - 210ms/step
Epoch 2964/5000
5/5 - 1s - loss: 5.4202e-05 - val_loss: 2.0808e-06 - 991ms/epoch - 198ms/s

Epoch 3044/5000
5/5 - 1s - loss: 7.4478e-05 - val_loss: 2.4856e-06 - 1s/epoch - 201ms/step
Epoch 3045/5000
5/5 - 1s - loss: 5.5051e-05 - val_loss: 6.6581e-06 - 1s/epoch - 207ms/step
Epoch 3046/5000
5/5 - 1s - loss: 5.8821e-05 - val_loss: 4.1951e-06 - 1s/epoch - 211ms/step
Epoch 3047/5000
5/5 - 1s - loss: 5.4834e-05 - val_loss: 1.6209e-06 - 1s/epoch - 202ms/step
Epoch 3048/5000
5/5 - 1s - loss: 6.3668e-05 - val_loss: 2.4907e-06 - 986ms/epoch - 197ms/step
Epoch 3049/5000
5/5 - 1s - loss: 6.5150e-05 - val_loss: 2.9079e-06 - 1s/epoch - 227ms/step
Epoch 3050/5000
5/5 - 1s - loss: 9.3904e-05 - val_loss: 2.2814e-06 - 1s/epoch - 203ms/step
Epoch 3051/5000
5/5 - 1s - loss: 9.6058e-05 - val_loss: 4.4863e-06 - 1s/epoch - 202ms/step
Epoch 3052/5000
5/5 - 1s - loss: 9.4647e-05 - val_loss: 5.0140e-06 - 1s/epoch - 208ms/step
Epoch 3053/5000
5/5 - 1s - loss: 6.9713e-05 - val_loss: 9.2020e-06 - 1s/epoch - 215ms/step
Epoch 3054/5000
5/5 - 1s - loss: 7.0959e-05 - val_loss: 2.3005e-06 - 1s/epoch - 211ms/s

Epoch 3133/5000
5/5 - 1s - loss: 7.3868e-05 - val_loss: 1.9308e-06 - 965ms/epoch - 193ms/step
Epoch 3134/5000
5/5 - 1s - loss: 6.2593e-05 - val_loss: 2.1827e-06 - 1s/epoch - 211ms/step
Epoch 3135/5000
5/5 - 1s - loss: 5.4702e-05 - val_loss: 1.7190e-06 - 936ms/epoch - 187ms/step
Epoch 3136/5000
5/5 - 1s - loss: 6.9263e-05 - val_loss: 4.7382e-06 - 1s/epoch - 202ms/step
Epoch 3137/5000
5/5 - 1s - loss: 9.5430e-05 - val_loss: 1.7313e-06 - 1s/epoch - 203ms/step
Epoch 3138/5000
5/5 - 1s - loss: 7.4382e-05 - val_loss: 1.0649e-05 - 936ms/epoch - 187ms/step
Epoch 3139/5000
5/5 - 1s - loss: 6.0856e-05 - val_loss: 1.2824e-05 - 987ms/epoch - 197ms/step
Epoch 3140/5000
5/5 - 1s - loss: 6.9961e-05 - val_loss: 8.6435e-06 - 961ms/epoch - 192ms/step
Epoch 3141/5000
5/5 - 1s - loss: 1.0562e-04 - val_loss: 1.7000e-06 - 1s/epoch - 213ms/step
Epoch 3142/5000
5/5 - 1s - loss: 6.4311e-05 - val_loss: 5.3368e-06 - 1000ms/epoch - 200ms/step
Epoch 3143/5000
5/5 - 1s - loss: 5.8766e-05 - val_loss: 1.6693e-06 - 98

Epoch 3222/5000
5/5 - 1s - loss: 8.0182e-05 - val_loss: 5.2023e-06 - 988ms/epoch - 198ms/step
Epoch 3223/5000
5/5 - 1s - loss: 6.1664e-05 - val_loss: 4.8570e-06 - 994ms/epoch - 199ms/step
Epoch 3224/5000
5/5 - 1s - loss: 8.7243e-05 - val_loss: 3.9235e-06 - 1s/epoch - 202ms/step
Epoch 3225/5000
5/5 - 1s - loss: 8.5957e-05 - val_loss: 1.7449e-05 - 1s/epoch - 201ms/step
Epoch 3226/5000
5/5 - 1s - loss: 1.0660e-04 - val_loss: 1.0015e-05 - 1s/epoch - 209ms/step
Epoch 3227/5000
5/5 - 1s - loss: 8.4084e-05 - val_loss: 1.4950e-05 - 1s/epoch - 207ms/step
Epoch 3228/5000
5/5 - 1s - loss: 6.4853e-05 - val_loss: 1.8107e-06 - 973ms/epoch - 195ms/step
Epoch 3229/5000
5/5 - 1s - loss: 6.3926e-05 - val_loss: 3.4463e-06 - 941ms/epoch - 188ms/step
Epoch 3230/5000
5/5 - 1s - loss: 6.6756e-05 - val_loss: 4.3613e-06 - 941ms/epoch - 188ms/step
Epoch 3231/5000
5/5 - 1s - loss: 6.3629e-05 - val_loss: 2.1461e-06 - 974ms/epoch - 195ms/step
Epoch 3232/5000
5/5 - 1s - loss: 5.6195e-05 - val_loss: 1.9561e-06 - 1s/

Epoch 3311/5000
5/5 - 1s - loss: 6.9581e-05 - val_loss: 2.8055e-06 - 920ms/epoch - 184ms/step
Epoch 3312/5000
5/5 - 1s - loss: 5.9422e-05 - val_loss: 2.8075e-06 - 962ms/epoch - 192ms/step
Epoch 3313/5000
5/5 - 1s - loss: 5.9158e-05 - val_loss: 6.4425e-06 - 1s/epoch - 212ms/step
Epoch 3314/5000
5/5 - 1s - loss: 6.2468e-05 - val_loss: 5.2502e-06 - 987ms/epoch - 197ms/step
Epoch 3315/5000
5/5 - 1s - loss: 5.6822e-05 - val_loss: 2.4364e-06 - 950ms/epoch - 190ms/step
Epoch 3316/5000
5/5 - 1s - loss: 6.1006e-05 - val_loss: 5.1222e-06 - 906ms/epoch - 181ms/step
Epoch 3317/5000
5/5 - 1s - loss: 5.7335e-05 - val_loss: 1.7435e-06 - 1s/epoch - 214ms/step
Epoch 3318/5000
5/5 - 1s - loss: 6.8751e-05 - val_loss: 1.4979e-06 - 1s/epoch - 219ms/step
Epoch 3319/5000
5/5 - 1s - loss: 7.7520e-05 - val_loss: 3.3093e-06 - 991ms/epoch - 198ms/step
Epoch 3320/5000
5/5 - 1s - loss: 5.8121e-05 - val_loss: 2.8059e-06 - 909ms/epoch - 182ms/step
Epoch 3321/5000
5/5 - 1s - loss: 5.3895e-05 - val_loss: 2.2112e-06 - 

Epoch 3400/5000
5/5 - 1s - loss: 5.4898e-05 - val_loss: 1.7909e-06 - 1s/epoch - 220ms/step
Epoch 3401/5000
5/5 - 1s - loss: 5.4957e-05 - val_loss: 2.0098e-06 - 1s/epoch - 202ms/step
Epoch 3402/5000
5/5 - 1s - loss: 5.5200e-05 - val_loss: 1.8689e-06 - 990ms/epoch - 198ms/step
Epoch 3403/5000
5/5 - 1s - loss: 5.3116e-05 - val_loss: 1.9158e-06 - 984ms/epoch - 197ms/step
Epoch 3404/5000
5/5 - 1s - loss: 5.2413e-05 - val_loss: 2.5169e-06 - 1s/epoch - 225ms/step
Epoch 3405/5000
5/5 - 1s - loss: 1.0395e-04 - val_loss: 2.4291e-06 - 1s/epoch - 228ms/step
Epoch 3406/5000
5/5 - 1s - loss: 6.9504e-05 - val_loss: 1.1446e-05 - 1s/epoch - 222ms/step
Epoch 3407/5000
5/5 - 1s - loss: 8.6245e-05 - val_loss: 1.7793e-06 - 1s/epoch - 225ms/step
Epoch 3408/5000
5/5 - 1s - loss: 5.8062e-05 - val_loss: 7.6198e-06 - 1s/epoch - 233ms/step
Epoch 3409/5000
5/5 - 1s - loss: 6.1340e-05 - val_loss: 1.6612e-06 - 1s/epoch - 219ms/step
Epoch 3410/5000
5/5 - 1s - loss: 5.5117e-05 - val_loss: 3.3424e-06 - 1s/epoch - 210m

Epoch 3490/5000
5/5 - 1s - loss: 5.8510e-05 - val_loss: 1.7866e-06 - 1s/epoch - 237ms/step
Epoch 3491/5000
5/5 - 1s - loss: 5.5510e-05 - val_loss: 2.0594e-06 - 1s/epoch - 223ms/step
Epoch 3492/5000
5/5 - 1s - loss: 5.7133e-05 - val_loss: 1.9632e-06 - 1s/epoch - 233ms/step
Epoch 3493/5000
5/5 - 1s - loss: 5.4079e-05 - val_loss: 2.1349e-06 - 1s/epoch - 216ms/step
Epoch 3494/5000
5/5 - 1s - loss: 5.4532e-05 - val_loss: 1.7511e-06 - 1s/epoch - 218ms/step
Epoch 3495/5000
5/5 - 1s - loss: 6.7857e-05 - val_loss: 1.6615e-06 - 1s/epoch - 240ms/step
Epoch 3496/5000
5/5 - 1s - loss: 5.8481e-05 - val_loss: 2.1284e-06 - 1s/epoch - 227ms/step
Epoch 3497/5000
5/5 - 1s - loss: 5.9993e-05 - val_loss: 2.0464e-06 - 1s/epoch - 228ms/step
Epoch 3498/5000
5/5 - 1s - loss: 5.5924e-05 - val_loss: 2.0154e-06 - 1s/epoch - 228ms/step
Epoch 3499/5000
5/5 - 1s - loss: 5.2137e-05 - val_loss: 1.9610e-06 - 1s/epoch - 219ms/step
Epoch 3500/5000
5/5 - 1s - loss: 5.3553e-05 - val_loss: 2.1661e-06 - 1s/epoch - 216ms/step

Epoch 3580/5000
5/5 - 1s - loss: 5.8933e-05 - val_loss: 4.7369e-06 - 1s/epoch - 227ms/step
Epoch 3581/5000
5/5 - 1s - loss: 5.5303e-05 - val_loss: 3.5418e-06 - 1s/epoch - 228ms/step
Epoch 3582/5000
5/5 - 1s - loss: 6.3179e-05 - val_loss: 1.5458e-06 - 1s/epoch - 245ms/step
Epoch 3583/5000
5/5 - 1s - loss: 5.4676e-05 - val_loss: 5.2321e-06 - 1s/epoch - 261ms/step
Epoch 3584/5000
5/5 - 1s - loss: 5.5091e-05 - val_loss: 2.5436e-06 - 1s/epoch - 234ms/step
Epoch 3585/5000
5/5 - 1s - loss: 5.3352e-05 - val_loss: 2.6261e-06 - 1s/epoch - 243ms/step
Epoch 3586/5000
5/5 - 1s - loss: 6.3092e-05 - val_loss: 2.8985e-06 - 1s/epoch - 231ms/step
Epoch 3587/5000
5/5 - 1s - loss: 7.2030e-05 - val_loss: 1.7008e-06 - 1s/epoch - 212ms/step
Epoch 3588/5000
5/5 - 1s - loss: 6.3742e-05 - val_loss: 3.3229e-06 - 1s/epoch - 210ms/step
Epoch 3589/5000
5/5 - 1s - loss: 6.2435e-05 - val_loss: 3.5181e-06 - 1s/epoch - 248ms/step
Epoch 3590/5000
5/5 - 1s - loss: 6.0078e-05 - val_loss: 1.4529e-06 - 1s/epoch - 235ms/step

Epoch 3670/5000
5/5 - 1s - loss: 7.1364e-05 - val_loss: 4.1176e-06 - 1s/epoch - 235ms/step
Epoch 3671/5000
5/5 - 1s - loss: 5.8010e-05 - val_loss: 5.1806e-06 - 1s/epoch - 221ms/step
Epoch 3672/5000
5/5 - 1s - loss: 5.5085e-05 - val_loss: 1.8370e-06 - 998ms/epoch - 200ms/step
Epoch 3673/5000
5/5 - 1s - loss: 5.3478e-05 - val_loss: 1.5902e-06 - 990ms/epoch - 198ms/step
Epoch 3674/5000
5/5 - 1s - loss: 5.2996e-05 - val_loss: 1.5669e-06 - 1s/epoch - 223ms/step
Epoch 3675/5000
5/5 - 1s - loss: 5.2385e-05 - val_loss: 1.9383e-06 - 1s/epoch - 228ms/step
Epoch 3676/5000
5/5 - 1s - loss: 7.1041e-05 - val_loss: 4.3861e-06 - 1s/epoch - 234ms/step
Epoch 3677/5000
5/5 - 1s - loss: 8.6041e-05 - val_loss: 1.8160e-06 - 1s/epoch - 210ms/step
Epoch 3678/5000
5/5 - 1s - loss: 8.9714e-05 - val_loss: 2.5750e-06 - 977ms/epoch - 195ms/step
Epoch 3679/5000
5/5 - 1s - loss: 6.3955e-05 - val_loss: 1.5262e-06 - 970ms/epoch - 194ms/step
Epoch 3680/5000
5/5 - 1s - loss: 7.0504e-05 - val_loss: 3.8625e-06 - 1s/epoch 

Epoch 3759/5000
5/5 - 1s - loss: 1.1456e-04 - val_loss: 1.4870e-06 - 1s/epoch - 200ms/step
Epoch 3760/5000
5/5 - 1s - loss: 1.6235e-04 - val_loss: 2.0529e-06 - 970ms/epoch - 194ms/step
Epoch 3761/5000
5/5 - 1s - loss: 1.2222e-04 - val_loss: 1.5523e-05 - 971ms/epoch - 194ms/step
Epoch 3762/5000
5/5 - 1s - loss: 9.3995e-05 - val_loss: 1.5740e-06 - 1s/epoch - 205ms/step
Epoch 3763/5000
5/5 - 1s - loss: 8.8601e-05 - val_loss: 3.5782e-06 - 1s/epoch - 215ms/step
Epoch 3764/5000
5/5 - 1s - loss: 6.7354e-05 - val_loss: 3.1922e-06 - 1s/epoch - 216ms/step
Epoch 3765/5000
5/5 - 1s - loss: 5.8266e-05 - val_loss: 1.4351e-06 - 948ms/epoch - 190ms/step
Epoch 3766/5000
5/5 - 1s - loss: 5.6125e-05 - val_loss: 2.3536e-06 - 909ms/epoch - 182ms/step
Epoch 3767/5000
5/5 - 1s - loss: 8.7313e-05 - val_loss: 6.3948e-06 - 1s/epoch - 206ms/step
Epoch 3768/5000
5/5 - 1s - loss: 1.8247e-04 - val_loss: 7.7996e-06 - 1s/epoch - 227ms/step
Epoch 3769/5000
5/5 - 1s - loss: 1.7205e-04 - val_loss: 1.3041e-05 - 1s/epoch 

Epoch 3849/5000
5/5 - 1s - loss: 6.9764e-05 - val_loss: 4.3907e-06 - 946ms/epoch - 189ms/step
Epoch 3850/5000
5/5 - 1s - loss: 8.8182e-05 - val_loss: 1.7839e-06 - 907ms/epoch - 181ms/step
Epoch 3851/5000
5/5 - 1s - loss: 5.7099e-05 - val_loss: 1.9048e-06 - 976ms/epoch - 195ms/step
Epoch 3852/5000
5/5 - 1s - loss: 6.3520e-05 - val_loss: 2.4917e-06 - 980ms/epoch - 196ms/step
Epoch 3853/5000
5/5 - 1s - loss: 5.3465e-05 - val_loss: 2.8322e-06 - 1s/epoch - 210ms/step
Epoch 3854/5000
5/5 - 1s - loss: 5.3384e-05 - val_loss: 2.2046e-06 - 963ms/epoch - 193ms/step
Epoch 3855/5000
5/5 - 1s - loss: 5.2510e-05 - val_loss: 1.4596e-06 - 1s/epoch - 217ms/step
Epoch 3856/5000
5/5 - 1s - loss: 5.2244e-05 - val_loss: 1.3881e-06 - 963ms/epoch - 193ms/step
Epoch 3857/5000
5/5 - 1s - loss: 5.1828e-05 - val_loss: 1.3565e-06 - 965ms/epoch - 193ms/step
Epoch 3858/5000
5/5 - 1s - loss: 5.1248e-05 - val_loss: 1.7560e-06 - 1s/epoch - 203ms/step
Epoch 3859/5000
5/5 - 1s - loss: 5.3371e-05 - val_loss: 1.3662e-06 - 

Epoch 3938/5000
5/5 - 1s - loss: 1.2369e-04 - val_loss: 1.6039e-06 - 1s/epoch - 202ms/step
Epoch 3939/5000
5/5 - 1s - loss: 6.4125e-05 - val_loss: 1.8128e-06 - 1s/epoch - 201ms/step
Epoch 3940/5000
5/5 - 1s - loss: 6.7319e-05 - val_loss: 3.9046e-06 - 898ms/epoch - 180ms/step
Epoch 3941/5000
5/5 - 1s - loss: 6.4666e-05 - val_loss: 3.1890e-06 - 959ms/epoch - 192ms/step
Epoch 3942/5000
5/5 - 1s - loss: 5.9338e-05 - val_loss: 6.9519e-06 - 1s/epoch - 237ms/step
Epoch 3943/5000
5/5 - 1s - loss: 5.8305e-05 - val_loss: 9.7538e-06 - 1s/epoch - 213ms/step
Epoch 3944/5000
5/5 - 1s - loss: 5.7790e-05 - val_loss: 6.5509e-06 - 1s/epoch - 201ms/step
Epoch 3945/5000
5/5 - 1s - loss: 5.5779e-05 - val_loss: 1.9389e-06 - 1s/epoch - 214ms/step
Epoch 3946/5000
5/5 - 1s - loss: 5.3446e-05 - val_loss: 1.9034e-06 - 1s/epoch - 219ms/step
Epoch 3947/5000
5/5 - 1s - loss: 5.6204e-05 - val_loss: 2.3347e-06 - 1s/epoch - 208ms/step
Epoch 3948/5000
5/5 - 1s - loss: 5.3057e-05 - val_loss: 4.4501e-06 - 1s/epoch - 211m

5/5 - 1s - loss: 6.3139e-05 - val_loss: 5.4262e-06 - 982ms/epoch - 196ms/step
Epoch 4028/5000
5/5 - 1s - loss: 5.9684e-05 - val_loss: 2.4306e-06 - 965ms/epoch - 193ms/step
Epoch 4029/5000
5/5 - 1s - loss: 9.5379e-05 - val_loss: 2.7479e-06 - 941ms/epoch - 188ms/step
Epoch 4030/5000
5/5 - 1s - loss: 1.2727e-04 - val_loss: 1.4738e-06 - 978ms/epoch - 196ms/step
Epoch 4031/5000
5/5 - 1s - loss: 7.9427e-05 - val_loss: 3.6871e-06 - 1s/epoch - 216ms/step
Epoch 4032/5000
5/5 - 1s - loss: 8.2326e-05 - val_loss: 2.8330e-06 - 1s/epoch - 203ms/step
Epoch 4033/5000
5/5 - 1s - loss: 6.7127e-05 - val_loss: 2.4028e-06 - 1s/epoch - 213ms/step
Epoch 4034/5000
5/5 - 1s - loss: 7.1517e-05 - val_loss: 1.4277e-06 - 1s/epoch - 207ms/step
Epoch 4035/5000
5/5 - 1s - loss: 5.5516e-05 - val_loss: 2.5394e-06 - 1s/epoch - 223ms/step
Epoch 4036/5000
5/5 - 1s - loss: 5.7892e-05 - val_loss: 2.0115e-06 - 1s/epoch - 247ms/step
Epoch 4037/5000
5/5 - 1s - loss: 5.2274e-05 - val_loss: 2.8552e-06 - 1s/epoch - 210ms/step
Epo

Epoch 4117/5000
5/5 - 1s - loss: 5.9648e-05 - val_loss: 2.2061e-06 - 996ms/epoch - 199ms/step
Epoch 4118/5000
5/5 - 1s - loss: 5.8660e-05 - val_loss: 1.5819e-06 - 1s/epoch - 201ms/step
Epoch 4119/5000
5/5 - 1s - loss: 5.3163e-05 - val_loss: 1.9197e-06 - 986ms/epoch - 197ms/step
Epoch 4120/5000
5/5 - 1s - loss: 6.0312e-05 - val_loss: 1.7184e-06 - 979ms/epoch - 196ms/step
Epoch 4121/5000
5/5 - 1s - loss: 7.5667e-05 - val_loss: 4.6228e-06 - 1s/epoch - 206ms/step
Epoch 4122/5000
5/5 - 1s - loss: 8.7794e-05 - val_loss: 2.0018e-06 - 1s/epoch - 225ms/step
Epoch 4123/5000
5/5 - 1s - loss: 6.4654e-05 - val_loss: 1.0378e-05 - 1s/epoch - 205ms/step
Epoch 4124/5000
5/5 - 1s - loss: 5.7425e-05 - val_loss: 1.7934e-06 - 933ms/epoch - 187ms/step
Epoch 4125/5000
5/5 - 1s - loss: 7.2668e-05 - val_loss: 1.8798e-06 - 960ms/epoch - 192ms/step
Epoch 4126/5000
5/5 - 1s - loss: 9.2458e-05 - val_loss: 2.3819e-06 - 1s/epoch - 215ms/step
Epoch 4127/5000
5/5 - 1s - loss: 6.9405e-05 - val_loss: 3.1285e-06 - 1s/epo

5/5 - 1s - loss: 5.6578e-05 - val_loss: 2.9520e-06 - 984ms/epoch - 197ms/step
Epoch 4207/5000
5/5 - 1s - loss: 5.7127e-05 - val_loss: 1.7881e-06 - 956ms/epoch - 191ms/step
Epoch 4208/5000
5/5 - 1s - loss: 7.6289e-05 - val_loss: 1.7078e-06 - 996ms/epoch - 199ms/step
Epoch 4209/5000
5/5 - 1s - loss: 5.9090e-05 - val_loss: 3.3285e-06 - 1s/epoch - 209ms/step
Epoch 4210/5000
5/5 - 1s - loss: 5.5946e-05 - val_loss: 4.0750e-06 - 960ms/epoch - 192ms/step
Epoch 4211/5000
5/5 - 1s - loss: 7.2002e-05 - val_loss: 3.2306e-06 - 992ms/epoch - 198ms/step
Epoch 4212/5000
5/5 - 1s - loss: 5.9760e-05 - val_loss: 2.8720e-06 - 982ms/epoch - 196ms/step
Epoch 4213/5000
5/5 - 1s - loss: 5.7437e-05 - val_loss: 1.9097e-06 - 1s/epoch - 204ms/step
Epoch 4214/5000
5/5 - 1s - loss: 5.4826e-05 - val_loss: 4.9254e-06 - 921ms/epoch - 184ms/step
Epoch 4215/5000
5/5 - 1s - loss: 5.4699e-05 - val_loss: 1.8615e-06 - 1s/epoch - 205ms/step
Epoch 4216/5000
5/5 - 1s - loss: 6.0274e-05 - val_loss: 1.6018e-06 - 976ms/epoch - 19

5/5 - 1s - loss: 9.3734e-05 - val_loss: 4.3704e-06 - 1s/epoch - 218ms/step
Epoch 4296/5000
5/5 - 1s - loss: 7.3853e-05 - val_loss: 1.9210e-05 - 1s/epoch - 210ms/step
Epoch 4297/5000
5/5 - 1s - loss: 7.7495e-05 - val_loss: 1.8393e-05 - 1s/epoch - 214ms/step
Epoch 4298/5000
5/5 - 1s - loss: 9.0750e-05 - val_loss: 1.4194e-06 - 1s/epoch - 223ms/step
Epoch 4299/5000
5/5 - 1s - loss: 7.2848e-05 - val_loss: 7.3723e-06 - 1s/epoch - 208ms/step
Epoch 4300/5000
5/5 - 1s - loss: 5.9545e-05 - val_loss: 1.5706e-06 - 1s/epoch - 221ms/step
Epoch 4301/5000
5/5 - 1s - loss: 5.9395e-05 - val_loss: 2.5811e-06 - 993ms/epoch - 199ms/step
Epoch 4302/5000
5/5 - 1s - loss: 6.5607e-05 - val_loss: 2.3766e-06 - 986ms/epoch - 197ms/step
Epoch 4303/5000
5/5 - 1s - loss: 7.0850e-05 - val_loss: 2.7294e-06 - 1s/epoch - 205ms/step
Epoch 4304/5000
5/5 - 1s - loss: 5.8193e-05 - val_loss: 2.4671e-06 - 1s/epoch - 217ms/step
Epoch 4305/5000
5/5 - 1s - loss: 1.2690e-04 - val_loss: 8.6732e-06 - 1s/epoch - 226ms/step
Epoch 430

Epoch 4385/5000
5/5 - 1s - loss: 1.3213e-04 - val_loss: 1.7274e-06 - 982ms/epoch - 196ms/step
Epoch 4386/5000
5/5 - 1s - loss: 8.1786e-05 - val_loss: 1.0124e-05 - 928ms/epoch - 186ms/step
Epoch 4387/5000
5/5 - 1s - loss: 6.8446e-05 - val_loss: 4.2994e-06 - 909ms/epoch - 182ms/step
Epoch 4388/5000
5/5 - 1s - loss: 6.6144e-05 - val_loss: 5.6683e-06 - 998ms/epoch - 200ms/step
Epoch 4389/5000
5/5 - 1s - loss: 6.0402e-05 - val_loss: 6.2153e-06 - 957ms/epoch - 191ms/step
Epoch 4390/5000
5/5 - 1s - loss: 5.5758e-05 - val_loss: 1.5000e-06 - 958ms/epoch - 192ms/step
Epoch 4391/5000
5/5 - 1s - loss: 5.3925e-05 - val_loss: 1.3442e-06 - 1s/epoch - 204ms/step
Epoch 4392/5000
5/5 - 1s - loss: 5.3754e-05 - val_loss: 1.5067e-06 - 1s/epoch - 206ms/step
Epoch 4393/5000
5/5 - 1s - loss: 5.2664e-05 - val_loss: 2.0558e-06 - 1s/epoch - 209ms/step
Epoch 4394/5000
5/5 - 1s - loss: 5.6313e-05 - val_loss: 1.7002e-06 - 1s/epoch - 217ms/step
Epoch 4395/5000
5/5 - 1s - loss: 6.2239e-05 - val_loss: 1.4305e-06 - 1s/

Epoch 4473/5000
5/5 - 1s - loss: 5.3875e-05 - val_loss: 2.1919e-06 - 908ms/epoch - 182ms/step
Epoch 4474/5000
5/5 - 1s - loss: 7.3884e-05 - val_loss: 2.0005e-06 - 1s/epoch - 208ms/step
Epoch 4475/5000
5/5 - 1s - loss: 5.4275e-05 - val_loss: 3.4441e-06 - 991ms/epoch - 198ms/step
Epoch 4476/5000
5/5 - 1s - loss: 6.5209e-05 - val_loss: 3.7239e-06 - 1s/epoch - 202ms/step
Epoch 4477/5000
5/5 - 1s - loss: 5.6929e-05 - val_loss: 1.6631e-06 - 956ms/epoch - 191ms/step
Epoch 4478/5000
5/5 - 1s - loss: 6.9203e-05 - val_loss: 1.7027e-06 - 1s/epoch - 210ms/step
Epoch 4479/5000
5/5 - 1s - loss: 6.2810e-05 - val_loss: 1.6442e-06 - 997ms/epoch - 199ms/step
Epoch 4480/5000
5/5 - 1s - loss: 5.5603e-05 - val_loss: 2.5602e-06 - 938ms/epoch - 188ms/step
Epoch 4481/5000
5/5 - 1s - loss: 6.0265e-05 - val_loss: 5.1916e-06 - 972ms/epoch - 194ms/step
Epoch 4482/5000
5/5 - 1s - loss: 5.5780e-05 - val_loss: 4.4169e-06 - 915ms/epoch - 183ms/step
Epoch 4483/5000
5/5 - 1s - loss: 5.2616e-05 - val_loss: 1.3163e-06 - 

Epoch 4561/5000
5/5 - 1s - loss: 5.7817e-05 - val_loss: 2.0924e-06 - 970ms/epoch - 194ms/step
Epoch 4562/5000
5/5 - 1s - loss: 5.7936e-05 - val_loss: 3.1718e-06 - 938ms/epoch - 188ms/step
Epoch 4563/5000
5/5 - 1s - loss: 8.9865e-05 - val_loss: 2.2890e-06 - 937ms/epoch - 187ms/step
Epoch 4564/5000
5/5 - 1s - loss: 8.9598e-05 - val_loss: 2.1859e-06 - 949ms/epoch - 190ms/step
Epoch 4565/5000
5/5 - 1s - loss: 6.5042e-05 - val_loss: 3.0317e-06 - 901ms/epoch - 180ms/step
Epoch 4566/5000
5/5 - 1s - loss: 7.2523e-05 - val_loss: 8.8016e-06 - 1s/epoch - 208ms/step
Epoch 4567/5000
5/5 - 1s - loss: 5.6857e-05 - val_loss: 3.0987e-06 - 921ms/epoch - 184ms/step
Epoch 4568/5000
5/5 - 1s - loss: 6.7156e-05 - val_loss: 4.4167e-06 - 1s/epoch - 201ms/step
Epoch 4569/5000
5/5 - 1s - loss: 5.6786e-05 - val_loss: 7.7212e-06 - 1s/epoch - 203ms/step
Epoch 4570/5000
5/5 - 1s - loss: 5.4219e-05 - val_loss: 1.5472e-06 - 974ms/epoch - 195ms/step
Epoch 4571/5000
5/5 - 1s - loss: 5.3170e-05 - val_loss: 1.3482e-06 - 

Epoch 4650/5000
5/5 - 1s - loss: 6.2576e-05 - val_loss: 2.8209e-06 - 1s/epoch - 201ms/step
Epoch 4651/5000
5/5 - 1s - loss: 6.0874e-05 - val_loss: 1.4003e-06 - 981ms/epoch - 196ms/step
Epoch 4652/5000
5/5 - 1s - loss: 5.4026e-05 - val_loss: 5.5322e-06 - 963ms/epoch - 193ms/step
Epoch 4653/5000
5/5 - 1s - loss: 5.3186e-05 - val_loss: 6.1711e-06 - 973ms/epoch - 195ms/step
Epoch 4654/5000
5/5 - 1s - loss: 5.7637e-05 - val_loss: 2.6483e-06 - 925ms/epoch - 185ms/step
Epoch 4655/5000
5/5 - 1s - loss: 5.5327e-05 - val_loss: 1.2758e-06 - 1s/epoch - 205ms/step
Epoch 4656/5000
5/5 - 1s - loss: 5.6043e-05 - val_loss: 1.3362e-06 - 985ms/epoch - 197ms/step
Epoch 4657/5000
5/5 - 1s - loss: 6.6242e-05 - val_loss: 2.7876e-06 - 992ms/epoch - 198ms/step
Epoch 4658/5000
5/5 - 1s - loss: 5.5441e-05 - val_loss: 3.0598e-06 - 1s/epoch - 211ms/step
Epoch 4659/5000
5/5 - 1s - loss: 6.9627e-05 - val_loss: 1.2994e-06 - 1s/epoch - 213ms/step
Epoch 4660/5000
5/5 - 1s - loss: 8.2415e-05 - val_loss: 1.6671e-06 - 1s/

Epoch 4739/5000
5/5 - 1s - loss: 5.4222e-05 - val_loss: 9.7099e-06 - 1s/epoch - 220ms/step
Epoch 4740/5000
5/5 - 1s - loss: 5.7074e-05 - val_loss: 3.5239e-06 - 1s/epoch - 207ms/step
Epoch 4741/5000
5/5 - 1s - loss: 6.5563e-05 - val_loss: 1.5309e-06 - 973ms/epoch - 195ms/step
Epoch 4742/5000
5/5 - 1s - loss: 6.5293e-05 - val_loss: 1.4127e-06 - 953ms/epoch - 191ms/step
Epoch 4743/5000
5/5 - 1s - loss: 6.8702e-05 - val_loss: 4.1547e-06 - 1s/epoch - 217ms/step
Epoch 4744/5000
5/5 - 1s - loss: 6.0308e-05 - val_loss: 3.7976e-06 - 1s/epoch - 217ms/step
Epoch 4745/5000
5/5 - 1s - loss: 5.5281e-05 - val_loss: 1.2901e-06 - 881ms/epoch - 176ms/step
Epoch 4746/5000
5/5 - 1s - loss: 5.3404e-05 - val_loss: 2.3407e-06 - 984ms/epoch - 197ms/step
Epoch 4747/5000
5/5 - 1s - loss: 1.4648e-04 - val_loss: 3.9851e-06 - 1s/epoch - 202ms/step
Epoch 4748/5000
5/5 - 1s - loss: 1.0440e-04 - val_loss: 2.1274e-05 - 1s/epoch - 202ms/step
Epoch 4749/5000
5/5 - 1s - loss: 1.0608e-04 - val_loss: 3.8192e-06 - 971ms/epo

Epoch 4828/5000
5/5 - 1s - loss: 5.9915e-05 - val_loss: 2.0211e-06 - 1s/epoch - 211ms/step
Epoch 4829/5000
5/5 - 1s - loss: 5.9414e-05 - val_loss: 1.4717e-06 - 995ms/epoch - 199ms/step
Epoch 4830/5000
5/5 - 1s - loss: 5.7300e-05 - val_loss: 3.1950e-06 - 974ms/epoch - 195ms/step
Epoch 4831/5000
5/5 - 1s - loss: 5.3655e-05 - val_loss: 1.3301e-06 - 976ms/epoch - 195ms/step
Epoch 4832/5000
5/5 - 1s - loss: 5.2757e-05 - val_loss: 2.6763e-06 - 983ms/epoch - 197ms/step
Epoch 4833/5000
5/5 - 1s - loss: 5.4305e-05 - val_loss: 2.8925e-06 - 987ms/epoch - 197ms/step
Epoch 4834/5000
5/5 - 1s - loss: 5.7544e-05 - val_loss: 2.5604e-06 - 928ms/epoch - 186ms/step
Epoch 4835/5000
5/5 - 1s - loss: 1.1855e-04 - val_loss: 1.9216e-06 - 1s/epoch - 203ms/step
Epoch 4836/5000
5/5 - 1s - loss: 8.5795e-05 - val_loss: 1.3312e-06 - 1s/epoch - 203ms/step
Epoch 4837/5000
5/5 - 1s - loss: 1.1296e-04 - val_loss: 1.1088e-05 - 941ms/epoch - 188ms/step
Epoch 4838/5000
5/5 - 1s - loss: 6.6155e-05 - val_loss: 1.1237e-05 - 

Epoch 4917/5000
5/5 - 1s - loss: 7.1108e-05 - val_loss: 2.6384e-06 - 970ms/epoch - 194ms/step
Epoch 4918/5000
5/5 - 1s - loss: 7.4880e-05 - val_loss: 6.1913e-06 - 991ms/epoch - 198ms/step
Epoch 4919/5000
5/5 - 1s - loss: 6.4854e-05 - val_loss: 2.3911e-06 - 979ms/epoch - 196ms/step
Epoch 4920/5000
5/5 - 1s - loss: 6.1908e-05 - val_loss: 2.9838e-06 - 990ms/epoch - 198ms/step
Epoch 4921/5000
5/5 - 1s - loss: 5.5019e-05 - val_loss: 1.5226e-06 - 1s/epoch - 220ms/step
Epoch 4922/5000
5/5 - 1s - loss: 8.9751e-05 - val_loss: 1.2428e-05 - 1s/epoch - 209ms/step
Epoch 4923/5000
5/5 - 1s - loss: 2.1241e-04 - val_loss: 1.8650e-06 - 1s/epoch - 210ms/step
Epoch 4924/5000
5/5 - 1s - loss: 3.4877e-04 - val_loss: 9.1619e-06 - 1s/epoch - 207ms/step
Epoch 4925/5000
5/5 - 1s - loss: 1.2586e-04 - val_loss: 2.8094e-06 - 1s/epoch - 206ms/step
Epoch 4926/5000
5/5 - 1s - loss: 1.4091e-04 - val_loss: 1.0653e-05 - 1s/epoch - 206ms/step
Epoch 4927/5000
5/5 - 1s - loss: 7.6911e-05 - val_loss: 1.2108e-05 - 971ms/epo

In [ ]:
# Callback Load
from keras.models import load_model, model_from_json
def load_keras_model(optimizer, loss):
    dirx = '/Users/hwangseokjun/Desktop/All/2024/[딥러닝]주가예측/models'
    os.chdir(dirx)
    json_file = open("LSTM_network.json",'r')
    loaded_model_json = json_file.read()
    json_file.close()
    model = model_from_json(loaded_model_json)
    model.compile(optimizer=optimizer, loss = loss)
    model.load_weights('LSTM_weights.h5')
    return model
LSTM_model = load_keras_model('adam','mse')

In [ ]:
# LSTM 평가
print("손실함수 : ", LSTM_model.evaluate(X_test,y_test))
pred_test = LSTM_model.predict(X_test)
plt.plot(pred_test,'r')
plt.plot(y_test,'g')